// @elementor/ui/index.js
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var React = require('react');
var MuiAccordion$1 = require('@mui/material/Accordion');
var MuiAccordionActions$1 = require('@mui/material/AccordionActions');
var MuiAccordionDetails = require('@mui/material/AccordionDetails');
var MuiAccordionSummary$1 = require('@mui/material/AccordionSummary');
var MuiSvgIcon = require('@mui/material/SvgIcon');
var MuiAlert = require('@mui/material/Alert');
var clsx = require('clsx');
var MuiIconButton$1 = require('@mui/material/IconButton');
var styles = require('@mui/material/styles');
var rtlPlugin = require('stylis-plugin-rtl');
var react = require('@emotion/react');
var createCache = require('@emotion/cache');
var stylis = require('stylis');
var useMediaQuery$1 = require('@mui/material/useMediaQuery');
var MuiButton$1 = require('@mui/material/Button');
var MuiAlertTitle = require('@mui/material/AlertTitle');
var MuiAppBar$1 = require('@mui/material/AppBar');
var MuiAutocomplete$1 = require('@mui/material/Autocomplete');
var MuiAvatar = require('@mui/material/Avatar');
var MuiAvatarGroup = require('@mui/material/AvatarGroup');
var MuiBackdrop = require('@mui/material/Backdrop');
var MuiBadge = require('@mui/material/Badge');
var MuiBottomNavigation = require('@mui/material/BottomNavigation');
var MuiBottomNavigationAction = require('@mui/material/BottomNavigationAction');
var MuiBox = require('@mui/material/Box');
var MuiBreadcrumbs = require('@mui/material/Breadcrumbs');
var MuiButtonBase$1 = require('@mui/material/ButtonBase');
var MuiButtonGroup$1 = require('@mui/material/ButtonGroup');
var MuiCard = require('@mui/material/Card');
var MuiCardActionArea = require('@mui/material/CardActionArea');
var MuiCardActions = require('@mui/material/CardActions');
var MuiCardContent = require('@mui/material/CardContent');
var MuiCardHeader = require('@mui/material/CardHeader');
var MuiCardMedia = require('@mui/material/CardMedia');
var MuiCheckbox = require('@mui/material/Checkbox');
var MuiChip$1 = require('@mui/material/Chip');
var MuiCircularProgress$1 = require('@mui/material/CircularProgress');
var MuiClickAwayListener = require('@mui/material/ClickAwayListener');
var MuiCollapse = require('@mui/material/Collapse');
var MuiContainer = require('@mui/material/Container');
var MuiCssBaseline = require('@mui/material/CssBaseline');
var DatePicker$1 = require('@mui/x-date-pickers/DatePicker');
var MuiInputAdornment = require('@mui/material/InputAdornment');
var MuiTextField = require('@mui/material/TextField');
var MuiDialog = require('@mui/material/Dialog');
var MuiDialogActions = require('@mui/material/DialogActions');
var MuiDialogContent$1 = require('@mui/material/DialogContent');
var MuiDialogContentText = require('@mui/material/DialogContentText');
var MuiToolbar = require('@mui/material/Toolbar');
var MuiStack = require('@mui/material/Stack');
var MuiDialogTitle$1 = require('@mui/material/DialogTitle');
var MuiDivider = require('@mui/material/Divider');
var MuiDrawer = require('@mui/material/Drawer');
var MuiFab = require('@mui/material/Fab');
var MuiFade = require('@mui/material/Fade');
var MuiFilledInput = require('@mui/material/FilledInput');
var MuiFormControl = require('@mui/material/FormControl');
var MuiFormControlLabel = require('@mui/material/FormControlLabel');
var MuiFormGroup = require('@mui/material/FormGroup');
var MuiFormHelperText$1 = require('@mui/material/FormHelperText');
var MuiFormLabel = require('@mui/material/FormLabel');
var MuiGrid = require('@mui/material/Grid');
var MuiGrow = require('@mui/material/Grow');
var MuiIcon = require('@mui/material/Icon');
var MuiImageList = require('@mui/material/ImageList');
var MuiImageListItem = require('@mui/material/ImageListItem');
var MuiImageListItemBar = require('@mui/material/ImageListItemBar');
var MuiInput = require('@mui/material/Input');
var MuiInputBase$1 = require('@mui/material/InputBase');
var MuiInputLabel = require('@mui/material/InputLabel');
var MuiLinearProgress = require('@mui/material/LinearProgress');
var MuiLink = require('@mui/material/Link');
var MuiList = require('@mui/material/List');
var MuiListItem$1 = require('@mui/material/ListItem');
var MuiListItemAvatar = require('@mui/material/ListItemAvatar');
var MuiListItemButton$1 = require('@mui/material/ListItemButton');
var MuiListItemIcon = require('@mui/material/ListItemIcon');
var MuiListItemSecondaryAction = require('@mui/material/ListItemSecondaryAction');
var MuiListItemText$1 = require('@mui/material/ListItemText');
var MuiListSubheader$1 = require('@mui/material/ListSubheader');
var LocalizationProvider$1 = require('@mui/x-date-pickers/LocalizationProvider');
var AdapterDayjs = require('@mui/x-date-pickers/AdapterDayjs');
var MuiMenu$1 = require('@mui/material/Menu');
var MuiMenuItem$1 = require('@mui/material/MenuItem');
var MuiMenuList = require('@mui/material/MenuList');
var MuiMobileStepper = require('@mui/material/MobileStepper');
var MuiModal = require('@mui/material/Modal');
var MuiNativeSelect = require('@mui/material/NativeSelect');
var MuiOutlinedInput = require('@mui/material/OutlinedInput');
var MuiPagination = require('@mui/material/Pagination');
var MuiPaginationItem = require('@mui/material/PaginationItem');
var MuiPaper$1 = require('@mui/material/Paper');
var MuiPopover = require('@mui/material/Popover');
var MuiPopper = require('@mui/material/Popper');
var MuiPortal = require('@mui/material/Portal');
var MuiRadio = require('@mui/material/Radio');
var MuiRadioGroup = require('@mui/material/RadioGroup');
var MuiRating = require('@mui/material/Rating');
var MuiScopedCssBaseline = require('@mui/material/ScopedCssBaseline');
var MuiSelect$1 = require('@mui/material/Select');
var MuiSkeleton = require('@mui/material/Skeleton');
var MuiSlide = require('@mui/material/Slide');
var MuiSlider = require('@mui/material/Slider');
var MuiSnackbar = require('@mui/material/Snackbar');
var MuiSnackbarContent = require('@mui/material/SnackbarContent');
var MuiSpeedDial = require('@mui/material/SpeedDial');
var MuiSpeedDialAction = require('@mui/material/SpeedDialAction');
var MuiSpeedDialIcon = require('@mui/material/SpeedDialIcon');
var MuiStep = require('@mui/material/Step');
var MuiStepButton = require('@mui/material/StepButton');
var MuiStepConnector = require('@mui/material/StepConnector');
var MuiStepContent = require('@mui/material/StepContent');
var MuiStepIcon$1 = require('@mui/material/StepIcon');
var MuiStepLabel$1 = require('@mui/material/StepLabel');
var MuiStepper$1 = require('@mui/material/Stepper');
var MuiSwipeableDrawer = require('@mui/material/SwipeableDrawer');
var MuiSwitch = require('@mui/material/Switch');
var MuiTab = require('@mui/material/Tab');
var MuiTabScrollButton = require('@mui/material/TabScrollButton');
var MuiTable = require('@mui/material/Table');
var MuiTableBody = require('@mui/material/TableBody');
var MuiTableCell = require('@mui/material/TableCell');
var MuiTableContainer = require('@mui/material/TableContainer');
var MuiTableFooter = require('@mui/material/TableFooter');
var MuiTableHead = require('@mui/material/TableHead');
var MuiTablePagination = require('@mui/material/TablePagination');
var MuiTableRow = require('@mui/material/TableRow');
var MuiTableSortLabel = require('@mui/material/TableSortLabel');
var MuiTabs = require('@mui/material/Tabs');
var MuiTextareaAutosize = require('@mui/material/TextareaAutosize');
var TimePicker$1 = require('@mui/x-date-pickers/TimePicker');
var MuiToggleButton$1 = require('@mui/material/ToggleButton');
var MuiToggleButtonGroup = require('@mui/material/ToggleButtonGroup');
var MuiTooltip$1 = require('@mui/material/Tooltip');
var MuiTypography = require('@mui/material/Typography');
var MuiZoom = require('@mui/material/Zoom');
var hooks = require('material-ui-popup-state/hooks');

function _interopDefaultLegacy (e) { return e && typeof e === 'object' && 'default' in e ? e : { 'default': e }; }

var React__default = /*#__PURE__*/_interopDefaultLegacy(React);
var MuiAccordion__default = /*#__PURE__*/_interopDefaultLegacy(MuiAccordion$1);
var MuiAccordionActions__default = /*#__PURE__*/_interopDefaultLegacy(MuiAccordionActions$1);
var MuiAccordionDetails__default = /*#__PURE__*/_interopDefaultLegacy(MuiAccordionDetails);
var MuiAccordionSummary__default = /*#__PURE__*/_interopDefaultLegacy(MuiAccordionSummary$1);
var MuiSvgIcon__default = /*#__PURE__*/_interopDefaultLegacy(MuiSvgIcon);
var MuiAlert__default = /*#__PURE__*/_interopDefaultLegacy(MuiAlert);
var clsx__default = /*#__PURE__*/_interopDefaultLegacy(clsx);
var MuiIconButton__default = /*#__PURE__*/_interopDefaultLegacy(MuiIconButton$1);
var rtlPlugin__default = /*#__PURE__*/_interopDefaultLegacy(rtlPlugin);
var createCache__default = /*#__PURE__*/_interopDefaultLegacy(createCache);
var useMediaQuery__default = /*#__PURE__*/_interopDefaultLegacy(useMediaQuery$1);
var MuiButton__default = /*#__PURE__*/_interopDefaultLegacy(MuiButton$1);
var MuiAlertTitle__default = /*#__PURE__*/_interopDefaultLegacy(MuiAlertTitle);
var MuiAppBar__default = /*#__PURE__*/_interopDefaultLegacy(MuiAppBar$1);
var MuiAutocomplete__default = /*#__PURE__*/_interopDefaultLegacy(MuiAutocomplete$1);
var MuiAvatar__default = /*#__PURE__*/_interopDefaultLegacy(MuiAvatar);
var MuiAvatarGroup__default = /*#__PURE__*/_interopDefaultLegacy(MuiAvatarGroup);
var MuiBackdrop__default = /*#__PURE__*/_interopDefaultLegacy(MuiBackdrop);
var MuiBadge__default = /*#__PURE__*/_interopDefaultLegacy(MuiBadge);
var MuiBottomNavigation__default = /*#__PURE__*/_interopDefaultLegacy(MuiBottomNavigation);
var MuiBottomNavigationAction__default = /*#__PURE__*/_interopDefaultLegacy(MuiBottomNavigationAction);
var MuiBox__default = /*#__PURE__*/_interopDefaultLegacy(MuiBox);
var MuiBreadcrumbs__default = /*#__PURE__*/_interopDefaultLegacy(MuiBreadcrumbs);
var MuiButtonBase__default = /*#__PURE__*/_interopDefaultLegacy(MuiButtonBase$1);
var MuiButtonGroup__default = /*#__PURE__*/_interopDefaultLegacy(MuiButtonGroup$1);
var MuiCard__default = /*#__PURE__*/_interopDefaultLegacy(MuiCard);
var MuiCardActionArea__default = /*#__PURE__*/_interopDefaultLegacy(MuiCardActionArea);
var MuiCardActions__default = /*#__PURE__*/_interopDefaultLegacy(MuiCardActions);
var MuiCardContent__default = /*#__PURE__*/_interopDefaultLegacy(MuiCardContent);
var MuiCardHeader__default = /*#__PURE__*/_interopDefaultLegacy(MuiCardHeader);
var MuiCardMedia__default = /*#__PURE__*/_interopDefaultLegacy(MuiCardMedia);
var MuiCheckbox__default = /*#__PURE__*/_interopDefaultLegacy(MuiCheckbox);
var MuiChip__default = /*#__PURE__*/_interopDefaultLegacy(MuiChip$1);
var MuiCircularProgress__default = /*#__PURE__*/_interopDefaultLegacy(MuiCircularProgress$1);
var MuiClickAwayListener__default = /*#__PURE__*/_interopDefaultLegacy(MuiClickAwayListener);
var MuiCollapse__default = /*#__PURE__*/_interopDefaultLegacy(MuiCollapse);
var MuiContainer__default = /*#__PURE__*/_interopDefaultLegacy(MuiContainer);
var MuiCssBaseline__default = /*#__PURE__*/_interopDefaultLegacy(MuiCssBaseline);
var MuiInputAdornment__default = /*#__PURE__*/_interopDefaultLegacy(MuiInputAdornment);
var MuiTextField__default = /*#__PURE__*/_interopDefaultLegacy(MuiTextField);
var MuiDialog__default = /*#__PURE__*/_interopDefaultLegacy(MuiDialog);
var MuiDialogActions__default = /*#__PURE__*/_interopDefaultLegacy(MuiDialogActions);
var MuiDialogContent__default = /*#__PURE__*/_interopDefaultLegacy(MuiDialogContent$1);
var MuiDialogContentText__default = /*#__PURE__*/_interopDefaultLegacy(MuiDialogContentText);
var MuiToolbar__default = /*#__PURE__*/_interopDefaultLegacy(MuiToolbar);
var MuiStack__default = /*#__PURE__*/_interopDefaultLegacy(MuiStack);
var MuiDialogTitle__default = /*#__PURE__*/_interopDefaultLegacy(MuiDialogTitle$1);
var MuiDivider__default = /*#__PURE__*/_interopDefaultLegacy(MuiDivider);
var MuiDrawer__default = /*#__PURE__*/_interopDefaultLegacy(MuiDrawer);
var MuiFab__default = /*#__PURE__*/_interopDefaultLegacy(MuiFab);
var MuiFade__default = /*#__PURE__*/_interopDefaultLegacy(MuiFade);
var MuiFilledInput__default = /*#__PURE__*/_interopDefaultLegacy(MuiFilledInput);
var MuiFormControl__default = /*#__PURE__*/_interopDefaultLegacy(MuiFormControl);
var MuiFormControlLabel__default = /*#__PURE__*/_interopDefaultLegacy(MuiFormControlLabel);
var MuiFormGroup__default = /*#__PURE__*/_interopDefaultLegacy(MuiFormGroup);
var MuiFormHelperText__default = /*#__PURE__*/_interopDefaultLegacy(MuiFormHelperText$1);
var MuiFormLabel__default = /*#__PURE__*/_interopDefaultLegacy(MuiFormLabel);
var MuiGrid__default = /*#__PURE__*/_interopDefaultLegacy(MuiGrid);
var MuiGrow__default = /*#__PURE__*/_interopDefaultLegacy(MuiGrow);
var MuiIcon__default = /*#__PURE__*/_interopDefaultLegacy(MuiIcon);
var MuiImageList__default = /*#__PURE__*/_interopDefaultLegacy(MuiImageList);
var MuiImageListItem__default = /*#__PURE__*/_interopDefaultLegacy(MuiImageListItem);
var MuiImageListItemBar__default = /*#__PURE__*/_interopDefaultLegacy(MuiImageListItemBar);
var MuiInput__default = /*#__PURE__*/_interopDefaultLegacy(MuiInput);
var MuiInputBase__default = /*#__PURE__*/_interopDefaultLegacy(MuiInputBase$1);
var MuiInputLabel__default = /*#__PURE__*/_interopDefaultLegacy(MuiInputLabel);
var MuiLinearProgress__default = /*#__PURE__*/_interopDefaultLegacy(MuiLinearProgress);
var MuiLink__default = /*#__PURE__*/_interopDefaultLegacy(MuiLink);
var MuiList__default = /*#__PURE__*/_interopDefaultLegacy(MuiList);
var MuiListItem__default = /*#__PURE__*/_interopDefaultLegacy(MuiListItem$1);
var MuiListItemAvatar__default = /*#__PURE__*/_interopDefaultLegacy(MuiListItemAvatar);
var MuiListItemButton__default = /*#__PURE__*/_interopDefaultLegacy(MuiListItemButton$1);
var MuiListItemIcon__default = /*#__PURE__*/_interopDefaultLegacy(MuiListItemIcon);
var MuiListItemSecondaryAction__default = /*#__PURE__*/_interopDefaultLegacy(MuiListItemSecondaryAction);
var MuiListItemText__default = /*#__PURE__*/_interopDefaultLegacy(MuiListItemText$1);
var MuiListSubheader__default = /*#__PURE__*/_interopDefaultLegacy(MuiListSubheader$1);
var MuiMenu__default = /*#__PURE__*/_interopDefaultLegacy(MuiMenu$1);
var MuiMenuItem__default = /*#__PURE__*/_interopDefaultLegacy(MuiMenuItem$1);
var MuiMenuList__default = /*#__PURE__*/_interopDefaultLegacy(MuiMenuList);
var MuiMobileStepper__default = /*#__PURE__*/_interopDefaultLegacy(MuiMobileStepper);
var MuiModal__default = /*#__PURE__*/_interopDefaultLegacy(MuiModal);
var MuiNativeSelect__default = /*#__PURE__*/_interopDefaultLegacy(MuiNativeSelect);
var MuiOutlinedInput__default = /*#__PURE__*/_interopDefaultLegacy(MuiOutlinedInput);
var MuiPagination__default = /*#__PURE__*/_interopDefaultLegacy(MuiPagination);
var MuiPaginationItem__default = /*#__PURE__*/_interopDefaultLegacy(MuiPaginationItem);
var MuiPaper__default = /*#__PURE__*/_interopDefaultLegacy(MuiPaper$1);
var MuiPopover__default = /*#__PURE__*/_interopDefaultLegacy(MuiPopover);
var MuiPopper__default = /*#__PURE__*/_interopDefaultLegacy(MuiPopper);
var MuiPortal__default = /*#__PURE__*/_interopDefaultLegacy(MuiPortal);
var MuiRadio__default = /*#__PURE__*/_interopDefaultLegacy(MuiRadio);
var MuiRadioGroup__default = /*#__PURE__*/_interopDefaultLegacy(MuiRadioGroup);
var MuiRating__default = /*#__PURE__*/_interopDefaultLegacy(MuiRating);
var MuiScopedCssBaseline__default = /*#__PURE__*/_interopDefaultLegacy(MuiScopedCssBaseline);
var MuiSelect__default = /*#__PURE__*/_interopDefaultLegacy(MuiSelect$1);
var MuiSkeleton__default = /*#__PURE__*/_interopDefaultLegacy(MuiSkeleton);
var MuiSlide__default = /*#__PURE__*/_interopDefaultLegacy(MuiSlide);
var MuiSlider__default = /*#__PURE__*/_interopDefaultLegacy(MuiSlider);
var MuiSnackbar__default = /*#__PURE__*/_interopDefaultLegacy(MuiSnackbar);
var MuiSnackbarContent__default = /*#__PURE__*/_interopDefaultLegacy(MuiSnackbarContent);
var MuiSpeedDial__default = /*#__PURE__*/_interopDefaultLegacy(MuiSpeedDial);
var MuiSpeedDialAction__default = /*#__PURE__*/_interopDefaultLegacy(MuiSpeedDialAction);
var MuiSpeedDialIcon__default = /*#__PURE__*/_interopDefaultLegacy(MuiSpeedDialIcon);
var MuiStep__default = /*#__PURE__*/_interopDefaultLegacy(MuiStep);
var MuiStepButton__default = /*#__PURE__*/_interopDefaultLegacy(MuiStepButton);
var MuiStepConnector__default = /*#__PURE__*/_interopDefaultLegacy(MuiStepConnector);
var MuiStepContent__default = /*#__PURE__*/_interopDefaultLegacy(MuiStepContent);
var MuiStepIcon__default = /*#__PURE__*/_interopDefaultLegacy(MuiStepIcon$1);
var MuiStepLabel__default = /*#__PURE__*/_interopDefaultLegacy(MuiStepLabel$1);
var MuiStepper__default = /*#__PURE__*/_interopDefaultLegacy(MuiStepper$1);
var MuiSwipeableDrawer__default = /*#__PURE__*/_interopDefaultLegacy(MuiSwipeableDrawer);
var MuiSwitch__default = /*#__PURE__*/_interopDefaultLegacy(MuiSwitch);
var MuiTab__default = /*#__PURE__*/_interopDefaultLegacy(MuiTab);
var MuiTabScrollButton__default = /*#__PURE__*/_interopDefaultLegacy(MuiTabScrollButton);
var MuiTable__default = /*#__PURE__*/_interopDefaultLegacy(MuiTable);
var MuiTableBody__default = /*#__PURE__*/_interopDefaultLegacy(MuiTableBody);
var MuiTableCell__default = /*#__PURE__*/_interopDefaultLegacy(MuiTableCell);
var MuiTableContainer__default = /*#__PURE__*/_interopDefaultLegacy(MuiTableContainer);
var MuiTableFooter__default = /*#__PURE__*/_interopDefaultLegacy(MuiTableFooter);
var MuiTableHead__default = /*#__PURE__*/_interopDefaultLegacy(MuiTableHead);
var MuiTablePagination__default = /*#__PURE__*/_interopDefaultLegacy(MuiTablePagination);
var MuiTableRow__default = /*#__PURE__*/_interopDefaultLegacy(MuiTableRow);
var MuiTableSortLabel__default = /*#__PURE__*/_interopDefaultLegacy(MuiTableSortLabel);
var MuiTabs__default = /*#__PURE__*/_interopDefaultLegacy(MuiTabs);
var MuiTextareaAutosize__default = /*#__PURE__*/_interopDefaultLegacy(MuiTextareaAutosize);
var MuiToggleButton__default = /*#__PURE__*/_interopDefaultLegacy(MuiToggleButton$1);
var MuiToggleButtonGroup__default = /*#__PURE__*/_interopDefaultLegacy(MuiToggleButtonGroup);
var MuiTooltip__default = /*#__PURE__*/_interopDefaultLegacy(MuiTooltip$1);
var MuiTypography__default = /*#__PURE__*/_interopDefaultLegacy(MuiTypography);
var MuiZoom__default = /*#__PURE__*/_interopDefaultLegacy(MuiZoom);

// Modified reason: modifying the default props.
const Accordion = React__default["default"].forwardRef((props, ref) => {
    return (React__default["default"].createElement(MuiAccordion__default["default"], { ...props, ref: ref }));
});
Accordion.defaultProps = {
    variant: 'outlined',
};

const AccordionActions = React__default["default"].forwardRef((props, ref) => {
    return (React__default["default"].createElement(MuiAccordionActions__default["default"], { ...props, ref: ref }));
});

const AccordionDetails = React__default["default"].forwardRef((props, ref) => {
    return (React__default["default"].createElement(MuiAccordionDetails__default["default"], { ...props, ref: ref }));
});

const SvgIcon = React__default["default"].forwardRef((props, ref) => {
    return (React__default["default"].createElement(MuiSvgIcon__default["default"], { ...props, ref: ref }));
});

const ArrowDropDownIcon$1 = React__default["default"].forwardRef((props, ref) => {
    return (React__default["default"].createElement(SvgIcon, { viewBox: "0 0 24 24", ...props, ref: ref },
        React__default["default"].createElement("path", { fillRule: "evenodd", clipRule: "evenodd", d: "M17.2929 9.29289C17.6834 8.90237 18.3166 8.90237 18.7071 9.29289C19.0976 9.68342 19.0976 10.3166 18.7071 10.7071L12.7071 16.7071C12.3166 17.0976 11.6834 17.0976 11.2929 16.7071L5.29289 10.7071C4.90237 10.3166 4.90237 9.68342 5.29289 9.29289C5.68342 8.90237 6.31658 8.90237 6.70711 9.29289L12 14.5858L17.2929 9.29289Z" })));
});
// Modified reason: modifying the default props.
const AccordionSummary = React__default["default"].forwardRef((props, ref) => {
    return (React__default["default"].createElement(MuiAccordionSummary__default["default"], { ...props, ref: ref }));
});
AccordionSummary.defaultProps = {
    expandIcon: React__default["default"].createElement(ArrowDropDownIcon$1, null),
};

// Modified reason: Material UI Typescript problem with the 'component' prop.
const IconButton = React__default["default"].forwardRef((props, ref) => {
    return (React__default["default"].createElement(MuiIconButton__default["default"], { ...props, ref: ref }));
});

/**
 * Do not edit directly
 * Generated on Wed, 10 Jan 2024 15:06:56 GMT
 */
const ColorCommonBlack = "#000000";
const ColorCommonWhite = "#ffffff";
const ColorGrey50 = "#f1f3f3";
const ColorGrey100 = "#d5d8dc";
const ColorGrey200 = "#babfc5";
const ColorGrey300 = "#9da5ae";
const ColorGrey400 = "#818a96";
const ColorGrey500 = "#69727d";
const ColorGrey600 = "#515962";
const ColorGrey700 = "#3f444b";
const ColorGrey800 = "#1f2124";
const ColorGrey900 = "#0c0d0e";
const ColorPink200 = "#f3bafd";
const ColorPink300 = "#f0abfc";
const ColorPink400 = "#eb8efb";
const ColorRed500 = "#ef4444";
const ColorRed600 = "#dc2626";
const ColorRed700 = "#b91c1c";
const ColorYellow400 = "#fbbf24";
const ColorYellow500 = "#f59e0b";
const ColorYellow600 = "#d97706";
const ColorYellow700 = "#bb5b1d";
const ColorYellow800 = "#b15211";
const ColorBlue500 = "#3b82f6";
const ColorBlue600 = "#2563eb";
const ColorBlue700 = "#1d4ed8";
const ColorGreen500 = "#10b981";
const ColorGreen600 = "#0a875a";
const ColorGreen700 = "#047857";
const ColorCyan300 = "#99f6e4";
const ColorCyan400 = "#5eead4";
const ColorCyan500 = "#2adfcd";
const ColorBurgundy700 = "#b51243";
const ColorBurgundy800 = "#93003f";
const ColorBurgundy900 = "#7e013b";

const MuiAccordion = {
    styleOverrides: {
        root: ({ theme }) => ({
            backgroundColor: theme.palette.background.default,
            '&:before': {
                // The accordion has an upper divider by default, which is not needed.
                content: 'none',
            },
            '&.Mui-expanded': {
                margin: 0,
            },
            '&.MuiAccordion-gutters + .MuiAccordion-root.MuiAccordion-gutters': {
                marginTop: theme.spacing(1),
                marginBottom: theme.spacing(0),
            },
            '&:not(.MuiAccordion-gutters) + .MuiAccordion-root:not(.MuiAccordion-gutters)': {
                borderTop: 0,
            },
            '&.Mui-disabled': {
                backgroundColor: theme.palette.background.default,
            },
        }),
    },
};

const MuiAccordionActions = {
    styleOverrides: {
        root: ({ theme }) => ({
            padding: theme.spacing(2),
        }),
    },
};

const MuiAccordionSummary = {
    styleOverrides: {
        root: () => ({
            // Matching the default minHeight of .MuiAccordionSummary.Mui-expanded
            minHeight: '64px',
        }),
        content: ({ theme }) => ({
            margin: theme.spacing(1, 0),
            '&.MuiAccordionSummary-content.Mui-expanded': {
                margin: theme.spacing(1, 0),
            },
        }),
    },
};

const MuiAppBar = {
    defaultProps: {
        elevation: 0,
        color: 'default',
    },
};

const MuiAutocomplete = {
    defaultProps: {
        slotProps: {
            paper: {
                elevation: 6,
            },
        },
    },
};

const MuiButton = {
    styleOverrides: {
        root: () => ({
            boxShadow: 'none',
            whiteSpace: 'nowrap',
            '&:hover': {
                boxShadow: 'none',
            },
            '& .MuiSvgIcon-root': {
                fill: 'currentColor',
            },
        }),
    },
    variants: [
        {
            props: { color: 'primary', variant: 'outlined' },
            style: () => ({
                // Temporary customizations until it will be decided in the design-system.
                color: '#C00BB9',
                borderColor: '#C00BB9',
                '&:hover': {
                    borderColor: '#C00BB9',
                },
            }),
        },
        {
            props: { color: 'primary', variant: 'text' },
            style: () => ({
                // Temporary customizations until it will be decided in the design-system.
                color: '#C00BB9',
            }),
        },
        {
            props: { color: 'global', variant: 'outlined' },
            style: () => ({
                // Temporary customizations until it will be decided in the design-system.
                color: '#17929B',
                borderColor: '#17929B',
                '&:hover': {
                    borderColor: '#17929B',
                },
            }),
        },
        {
            props: { color: 'global', variant: 'text' },
            style: () => ({
                // Temporary customizations until it will be decided in the design-system.
                color: '#17929B',
            }),
        },
    ],
};

const MuiButtonBase = {
    defaultProps: {
        disableRipple: true,
    },
    styleOverrides: {
        root: () => ({
            '&.MuiButtonBase-root.Mui-focusVisible': {
                // Temp customization for accessibility (using 'border' because 'outline' is not working properly in Tabs).
                boxShadow: '0 0 0 1px inset',
            },
            // The CircularProgress component used as a loader of the buttons.
            '.MuiCircularProgress-root': {
                fontSize: 'inherit',
            },
        }),
    },
};

const themePaletteSemanticColors = [
    'primary',
    'secondary',
    'error',
    'warning',
    'info',
    'success',
    'accent',
    'global',
    'promotion',
];
const LINK_PSEUDO_SELECTORS = '&:hover,&:focus,&:active,&:visited';

const MuiButtonGroup = {
    defaultProps: {
        disableRipple: true,
    },
    styleOverrides: {
        root: () => ({
            boxShadow: 'none',
            '&:hover': {
                boxShadow: 'none',
            },
        }),
    },
    variants: themePaletteSemanticColors.map((color) => ({
        props: { variant: 'contained', color },
        // Fixing a Material UI bug when one of the group children is wrapped with another element (e.g. a tooltip).
        style: ({ theme }) => {
            return {
                // Removing the border of the grouped buttons for supporting additional children types.
                '& .MuiButtonGroup-grouped:not(:last-of-type), & .MuiButtonGroup-grouped:not(:last-of-type).Mui-disabled': {
                    borderRight: 0,
                },
                // Supporting additional children types, when a button can be inside another element (e.g. a tooltip).
                '& .MuiButtonGroup-grouped:not(:last-child), & > *:not(:last-child) .MuiButtonGroup-grouped': {
                    borderRight: `1px solid ${theme.palette[color].dark}`,
                },
                // Supporting additional children types, when a disabled button can be inside another element (e.g. a tooltip).
                '& .MuiButtonGroup-grouped:not(:last-child).Mui-disabled, & > *:not(:last-child) .MuiButtonGroup-grouped.Mui-disabled': {
                    borderRight: `1px solid ${theme.palette.action.disabled}`,
                },
            };
        },
    })),
};

const MuiChip = {
    variants: [
        {
            props: { color: 'primary', variant: 'outlined' },
            style: () => ({
                // Temporary customizations until it will be decided in the design-system.
                color: '#C00BB9',
                borderColor: '#C00BB9',
                '& .MuiChip-deleteIcon': {
                    color: '#D355CE',
                    '&:hover': {
                        color: '#C00BB9',
                    },
                },
            }),
        },
        {
            props: { color: 'global', variant: 'outlined' },
            style: () => ({
                // Temporary customizations until it will be decided in the design-system.
                color: '#17929B',
                borderColor: '#17929B',
                '& .MuiChip-deleteIcon': {
                    color: '#5DB3B9',
                    '&:hover': {
                        color: '#17929B',
                    },
                },
            }),
        },
    ],
};

const MuiCircularProgress = {
    defaultProps: {
        color: 'inherit',
        /**
         * Customizing to 1em for easy customization by other components that wrap the CircularProgress.
         * Setting the default size value in the styleOverrides fontSize property.
         */
        size: '1em',
    },
    styleOverrides: {
        root: ({ theme }) => ({
            // Temporarily reading the value from the spacing, until we'll find a better solution.
            fontSize: theme.spacing(5),
        }),
    },
};

const MuiDialogContent = {
    styleOverrides: {
        dividers: () => ({
            '&:last-child': {
                // Fixing Material UI bug - when the divider is the last child it should not be shown.
                borderBottom: 'none',
            },
        }),
    },
};

const MuiDialogTitle = {
    styleOverrides: {
        root: ({ theme }) => ({
            fontWeight: theme.typography.fontWeightBold,
        }),
    },
};

const MuiFormHelperText = {
    styleOverrides: {
        root: ({ theme }) => ({
            color: theme.palette.text.tertiary,
            margin: theme.spacing(0.5, 0, 0),
        }),
    },
};

const MuiIconButton = {
    variants: [
        {
            props: { color: 'primary' },
            style: () => ({
                // Temporary customizations until it will be decided in the design-system.
                color: '#C00BB9',
            }),
        },
        {
            props: { color: 'global' },
            style: () => ({
                // Temporary customizations until it will be decided in the design-system.
                color: '#17929B',
            }),
        },
        {
            props: { edge: 'start', size: 'small' },
            style: ({ theme }) => ({
                marginLeft: theme.spacing(-1.5),
            }),
        },
        {
            props: { edge: 'end', size: 'small' },
            style: ({ theme }) => ({
                marginRight: theme.spacing(-1.5),
            }),
        },
        {
            props: { edge: 'start', size: 'large' },
            style: ({ theme }) => ({
                marginLeft: theme.spacing(-2),
            }),
        },
        {
            props: { edge: 'end', size: 'large' },
            style: ({ theme }) => ({
                marginRight: theme.spacing(-2),
            }),
        },
    ],
};

const MuiInputBase = {
    styleOverrides: {
        input: () => ({
            // Setting the background color to prevent global style overrides.
            '.MuiInputBase-root.Mui-disabled &': {
                backgroundColor: 'initial',
            },
        }),
    },
};

const MuiListItem = {
    styleOverrides: {
        root: ({ theme }) => ({
            color: theme.palette.text.primary,
            // Setting the pseudo-classes color to prevent global style overrides when the MenuItem is an "a" tag.
            'a&': {
                [LINK_PSEUDO_SELECTORS]: {
                    color: theme.palette.text.primary,
                },
            },
        }),
    },
};

const MuiListItemButton = {
    styleOverrides: {
        root: ({ theme }) => ({
            color: theme.palette.text.primary,
            // Setting the pseudo-classes color to prevent global style overrides when the MenuItem is an "a" tag.
            'a&': {
                [LINK_PSEUDO_SELECTORS]: {
                    color: theme.palette.text.primary,
                },
            },
        }),
    },
};

const MuiListItemText = {
    styleOverrides: {
        root: ({ theme }) => ({
            color: theme.palette.text.primary,
        }),
    },
};

// TODO: update the relevant components when the paper background image is changed.
const PAPER_BACKGROUND_IMAGE = 'linear-gradient(rgba(255, 255, 255, 0.12), rgba(255, 255, 255, 0.12))';
/**
 * The purpose of this empty configuration object is to make sure that all components that uses the PAPER_BACKGROUND_IMAGE const,
 * Should also be adjusted if the paper background image is changed.
 */
const MuiPaper = {};

const MuiListSubheader = {
    styleOverrides: {
        root: ({ theme }) => ({
            // The background color should be the same as the drop-downs background color.
            backgroundImage: PAPER_BACKGROUND_IMAGE,
            // Temp value until there will be a token in the design system.
            lineHeight: '36px',
            color: theme.palette.text.tertiary,
            fontSize: theme.typography.caption.fontSize,
            fontWeight: theme.typography.caption.fontWeight,
            fontFamily: theme.typography.caption.fontFamily,
            letterSpacing: theme.typography.caption.letterSpacing,
        }),
    },
};

const MuiMenu = {
    defaultProps: {
        elevation: 6,
    },
};

const MuiMenuItem = {
    styleOverrides: {
        root: ({ theme }) => ({
            // Setting the pseudo-classes color to prevent global style overrides when the MenuItem is an "a" tag.
            'a&': {
                [LINK_PSEUDO_SELECTORS]: {
                    color: theme.palette.text.primary,
                },
            },
        }),
    },
};

const MuiSelect = {
    styleOverrides: {
        nativeInput: () => ({
            // Setting the background color and opacity to prevent global style overrides.
            '.MuiInputBase-root.Mui-disabled &': {
                backgroundColor: 'initial',
                opacity: 0,
            },
        }),
    },
};

const MuiStepIcon = {
    styleOverrides: {
        root: ({ theme }) => ({
            '&:not(.Mui-active) .MuiStepIcon-text': {
                fill: theme.palette.common.white,
            },
        }),
    },
};

const MuiStepLabel = {
    styleOverrides: {
        root: () => ({
            alignItems: 'flex-start',
        }),
    },
};

const MuiStepper = {
    styleOverrides: {
        root: () => ({
            '& .MuiStepLabel-root': {
                alignItems: 'center',
            },
        }),
    },
};

const MuiToggleButton = {
    variants: [
        {
            props: { color: 'primary' },
            style: () => ({
                // Temporary customizations until it will be decided in the design-system.
                '&.MuiToggleButton-root.Mui-selected': {
                    color: '#C00BB9',
                },
            }),
        },
        {
            props: { color: 'global' },
            style: () => ({
                // Temporary customizations until it will be decided in the design-system.
                '&.MuiToggleButton-root.Mui-selected': {
                    color: '#17929B',
                },
            }),
        },
    ],
};

const MuiTooltip = {
    defaultProps: {
        arrow: true,
    },
};

var components = {
    MuiAccordion,
    MuiAccordionActions,
    MuiAccordionSummary,
    MuiAppBar,
    MuiAutocomplete,
    MuiButton,
    MuiButtonBase,
    MuiButtonGroup,
    MuiChip,
    MuiCircularProgress,
    MuiDialogContent,
    MuiDialogTitle,
    MuiFormHelperText,
    MuiIconButton,
    MuiInputBase,
    MuiListItem,
    MuiListItemButton,
    MuiListItemText,
    MuiListSubheader,
    MuiMenu,
    MuiMenuItem,
    MuiPaper,
    MuiSelect,
    MuiStepIcon,
    MuiStepLabel,
    MuiStepper,
    MuiToggleButton,
    MuiTooltip,
};

const themeBaseConfig = {
    components,
    typography: {
        button: {
            textTransform: 'none',
        },
        subtitle1: {
            lineHeight: 1.3,
        },
        subtitle2: {
            lineHeight: 1.3,
        },
    },
};

const lightTheme = styles.createTheme({
    ...themeBaseConfig,
    palette: {
        mode: 'light',
        primary: {
            main: ColorPink300,
            light: ColorPink200,
            dark: ColorPink400,
            contrastText: ColorGrey900,
        },
        secondary: {
            main: ColorGrey600,
            light: ColorGrey500,
            dark: ColorGrey700,
            contrastText: ColorCommonWhite,
        },
        grey: {
            50: ColorGrey50,
            100: ColorGrey100,
            200: ColorGrey200,
            300: ColorGrey300,
            400: ColorGrey400,
            500: ColorGrey500,
            600: ColorGrey600,
            700: ColorGrey700,
            800: ColorGrey800,
            900: ColorGrey900,
        },
        text: {
            primary: ColorGrey900,
            secondary: ColorGrey700,
            tertiary: ColorGrey500,
            disabled: ColorGrey300,
        },
        background: {
            paper: ColorCommonWhite,
            default: ColorCommonWhite,
        },
        success: {
            main: ColorGreen600,
            light: ColorGreen500,
            dark: ColorGreen700,
            contrastText: ColorCommonWhite,
        },
        error: {
            main: ColorRed600,
            light: ColorRed500,
            dark: ColorRed700,
            contrastText: ColorCommonWhite,
        },
        warning: {
            main: ColorYellow700,
            light: ColorYellow600,
            dark: ColorYellow800,
            contrastText: ColorCommonWhite,
        },
        info: {
            main: ColorBlue600,
            light: ColorBlue500,
            dark: ColorBlue700,
            contrastText: ColorCommonWhite,
        },
        global: {
            main: ColorCyan400,
            light: ColorCyan300,
            dark: ColorCyan500,
            contrastText: ColorGrey900,
        },
        // TODO: the accent values should be updated as part of a deprecation process.
        accent: {
            main: ColorBurgundy800,
            light: ColorBurgundy700,
            dark: ColorBurgundy900,
            contrastText: ColorCommonWhite,
        },
        promotion: {
            main: ColorBurgundy800,
            light: ColorBurgundy700,
            dark: ColorBurgundy900,
            contrastText: ColorCommonWhite,
        },
    },
});

const darkTheme = styles.createTheme({
    ...themeBaseConfig,
    palette: {
        mode: 'dark',
        primary: {
            main: ColorPink300,
            light: ColorPink200,
            dark: ColorPink400,
            contrastText: ColorGrey900,
        },
        secondary: {
            main: ColorGrey300,
            light: ColorGrey200,
            dark: ColorGrey400,
            contrastText: ColorGrey900,
        },
        grey: {
            50: ColorGrey50,
            100: ColorGrey100,
            200: ColorGrey200,
            300: ColorGrey300,
            400: ColorGrey400,
            500: ColorGrey500,
            600: ColorGrey600,
            700: ColorGrey700,
            800: ColorGrey800,
            900: ColorGrey900,
        },
        text: {
            primary: ColorCommonWhite,
            secondary: ColorGrey200,
            tertiary: ColorGrey300,
            disabled: ColorGrey600,
        },
        background: {
            paper: ColorGrey900,
            default: ColorGrey800,
        },
        success: {
            main: ColorGreen600,
            light: ColorGreen500,
            dark: ColorGreen700,
            contrastText: ColorCommonWhite,
        },
        error: {
            main: ColorRed600,
            light: ColorRed500,
            dark: ColorRed700,
            contrastText: ColorCommonWhite,
        },
        warning: {
            main: ColorYellow500,
            light: ColorYellow400,
            dark: ColorYellow800,
            contrastText: ColorCommonBlack,
        },
        info: {
            main: ColorBlue600,
            light: ColorBlue500,
            dark: ColorBlue700,
            contrastText: ColorCommonWhite,
        },
        global: {
            main: ColorCyan400,
            light: ColorCyan300,
            dark: ColorCyan500,
            contrastText: ColorGrey900,
        },
        // TODO: the accent values should be updated as part of a deprecation process.
        accent: {
            main: ColorBurgundy800,
            light: ColorBurgundy700,
            dark: ColorBurgundy900,
            contrastText: ColorCommonWhite,
        },
        promotion: {
            main: ColorBurgundy800,
            light: ColorBurgundy700,
            dark: ColorBurgundy900,
            contrastText: ColorCommonWhite,
        },
    },
});

const DirectionContext = React__default["default"].createContext(false);
/**
 * @see https://mui.com/material-ui/guides/right-to-left/
 */
const cacheRtl = createCache__default["default"]({
    // Define class names prefix.
    key: 'eui-rtl',
    stylisPlugins: [stylis.prefixer, rtlPlugin__default["default"]],
});
const DirectionCacheProvider = (props) => {
    if (!props.isRTL) {
        return React__default["default"].createElement(React__default["default"].Fragment, null, props.children);
    }
    return React__default["default"].createElement(react.CacheProvider, { value: cacheRtl }, props.children);
};
const DirectionProvider = (props) => {
    const isRTL = !!props.rtl;
    return (React__default["default"].createElement(DirectionContext.Provider, { value: isRTL },
        React__default["default"].createElement(DirectionCacheProvider, { isRTL: isRTL }, props.children)));
};

const ThemeProvider = ({ colorScheme = 'auto', children }) => {
    const rtl = React.useContext(DirectionContext);
    const prefersDarkMode = useMediaQuery__default["default"]('(prefers-color-scheme: dark)');
    const theme = React.useMemo(() => {
        const themeData = ((colorScheme === 'auto' && prefersDarkMode) || colorScheme === 'dark') ? darkTheme : lightTheme;
        if (rtl) {
            return styles.createTheme(themeData, { direction: 'rtl' });
        }
        return themeData;
    }, [rtl, colorScheme, prefersDarkMode]);
    return (React__default["default"].createElement(styles.ThemeProvider, { theme: theme }, children));
};

const createSlots = (componentName, componentSlots) => {
    const slots = {};
    const classNames = {};
    componentSlots.forEach((slot) => {
        classNames[slot] = `Mui${componentName}-${slot}`;
        slots[slot] = {
            slot,
            name: `Mui${componentName}`,
        };
    });
    return {
        slots,
        classNames,
    };
};

// Must have a default export to match all other components structure, for the global index.js to work.
var index = {};

const XIcon = React__default["default"].forwardRef((props, ref) => {
    return (React__default["default"].createElement(SvgIcon, { viewBox: "0 0 24 24", ...props, ref: ref },
        React__default["default"].createElement("path", { fillRule: "evenodd", clipRule: "evenodd", d: "M18.5303 5.46967C18.8232 5.76256 18.8232 6.23744 18.5303 6.53033L6.53033 18.5303C6.23744 18.8232 5.76256 18.8232 5.46967 18.5303C5.17678 18.2374 5.17678 17.7626 5.46967 17.4697L17.4697 5.46967C17.7626 5.17678 18.2374 5.17678 18.5303 5.46967Z" }),
        React__default["default"].createElement("path", { fillRule: "evenodd", clipRule: "evenodd", d: "M5.46967 5.46967C5.76256 5.17678 6.23744 5.17678 6.53033 5.46967L18.5303 17.4697C18.8232 17.7626 18.8232 18.2374 18.5303 18.5303C18.2374 18.8232 17.7626 18.8232 17.4697 18.5303L5.46967 6.53033C5.17678 6.23744 5.17678 5.76256 5.46967 5.46967Z" })));
});
const { slots: slots$5, classNames: classNames$5 } = createSlots('CloseButton', ['root', 'icon']);
const Root$4 = styles.styled(IconButton, slots$5.root)({});
const Icon$1 = styles.styled(XIcon, slots$5.icon)({});
const CloseButton = React__default["default"].forwardRef((inProps, ref) => {
    const props = styles.useThemeProps({ props: inProps, name: slots$5.root.name });
    const { slotProps = {}, ...rootProps } = props;
    return (React__default["default"].createElement(Root$4, { ...rootProps, size: "small", ref: ref, className: clsx__default["default"]([[classNames$5.root, rootProps.className]]), ownerState: props },
        React__default["default"].createElement(Icon$1, { ...slotProps.icon, className: clsx__default["default"]([classNames$5.icon, slotProps.icon?.className]), ownerState: props })));
});
CloseButton.defaultProps = {
    'aria-label': 'close',
    color: 'default',
};

const StyledAlert = styles.styled(MuiAlert__default["default"])(({ theme, severity, color, variant }) => {
    const variantsStyle = getVariantsStyle(severity, color, variant, theme);
    return {
        padding: theme.spacing(1.5, 2),
        '& .MuiAlert-message': {
            width: '100%',
            padding: 0,
            minHeight: '30.75px',
            display: 'flex',
            flexDirection: 'row',
            flexWrap: 'wrap',
            gap: theme.spacing(1.5),
        },
        '& .MuiAlertTitle-root': {
            marginBottom: 0,
            lineHeight: 'inherit',
            marginRight: theme.spacing(0.5),
            ...theme.typography.subtitle2,
            marginTop: 0,
        },
        '& .MuiAlert-icon': {
            padding: 0,
            paddingTop: theme.spacing(0.5),
        },
        '& .MuiAlert-action': {
            padding: 0,
            marginLeft: theme.spacing(1),
        },
        '&.MuiAlert-filledWarning': {
            color: theme.palette.common.white,
        },
        ...variantsStyle,
    };
});
const { slots: slots$4, classNames: classNames$4 } = createSlots('Alert', ['actions', 'content']);
const StyledAlertContent = styles.styled('div', slots$4.content)(() => ({
    flexGrow: 1,
    paddingTop: '6px',
}));
const StyledAlertContentInnerContainer = styles.styled('div', slots$4.content)(({ theme }) => ({
    alignItems: 'center',
    display: 'flex',
    flexWrap: 'wrap',
    gap: theme.spacing(0.25),
    maxWidth: '800px',
}));
const AlertContent = ({ children, ...props }) => {
    return (React__default["default"].createElement(StyledAlertContent, { ...props },
        React__default["default"].createElement(StyledAlertContentInnerContainer, null, children)));
};
const StyledAlertActions = styles.styled('div')(({ theme }) => ({
    display: 'flex',
    alignItems: 'flex-start',
    flexWrap: 'wrap',
    gap: theme.spacing(1),
}));
// Modified reason: Allowing multiple actions with responsive layout.
const Alert = React__default["default"].forwardRef(({ onClose, action, secondaryAction, children, ...props }, ref) => {
    const hasActions = Boolean(action || secondaryAction);
    return (React__default["default"].createElement(StyledAlert, { iconMapping: {
            success: React__default["default"].createElement(SuccessIcon, null),
            error: React__default["default"].createElement(ErrorIcon, null),
            info: React__default["default"].createElement(InfoIcon, null),
            warning: React__default["default"].createElement(WarningIcon, null),
        }, ...props, ref: ref, action: onClose
            ? (React__default["default"].createElement(CloseButton, { color: "inherit", onClick: onClose, slotProps: {
                    icon: {
                        fontSize: 'small',
                    },
                }, title: props.closeText, "aria-label": props.closeText }))
            : false },
        React__default["default"].createElement(AlertContent, { className: classNames$4.content }, children),
        hasActions && (React__default["default"].createElement(StyledAlertActions, { className: classNames$4.actions },
            secondaryAction,
            action))));
});
Alert.defaultProps = {
    closeText: 'Close',
    severity: 'success',
};
function getVariantsStyle(severity, color, variant, theme) {
    // In Mui, the color prop has a higher priority than the severity prop.
    const semanticColor = color || severity;
    if (!semanticColor) {
        return {};
    }
    if (variant === 'filled') {
        return {
            '& .MuiButton-containedInherit': {
                color: theme.palette[semanticColor].main,
                backgroundColor: 'rgba(255, 255, 255, 1)',
                '&:hover': {
                    backgroundColor: 'rgba(255, 255, 255, .96)',
                },
            },
            '& .MuiButton-outlinedInherit:hover': {
                backgroundColor: theme.palette[semanticColor].dark,
            },
            // Setting the pseudo-classes color to prevent the override of global CSS with higher specificity than the Button component.
            '& a.MuiButtonBase-root.MuiButton-containedInherit': {
                [LINK_PSEUDO_SELECTORS]: {
                    color: theme.palette[semanticColor].main,
                },
            },
        };
    }
    return {
        '&.MuiAlert-root': {
            color: theme.palette.text.secondary,
        },
        '& .MuiCloseButton-root': {
            color: theme.palette.action.active,
        },
        '& .MuiButton-containedInherit': {
            color: theme.palette[semanticColor].contrastText,
            backgroundColor: theme.palette[semanticColor].main,
            '&:hover': {
                backgroundColor: theme.palette[semanticColor].dark,
            },
        },
        '& .MuiButton-outlinedInherit': {
            borderColor: theme.palette[semanticColor].main,
            color: theme.palette[semanticColor].main,
            '&:hover': {
                backgroundColor: styles.alpha(theme.palette[semanticColor].main, 0.08),
            },
        },
        // Setting the pseudo-classes color to prevent the override of global CSS with higher specificity than the Button component.
        '& a.MuiButtonBase-root.MuiButton-containedInherit': {
            [LINK_PSEUDO_SELECTORS]: {
                color: theme.palette[semanticColor].contrastText,
            },
        },
        '& a.MuiButtonBase-root.MuiButton-outlinedInherit': {
            [LINK_PSEUDO_SELECTORS]: {
                color: theme.palette[semanticColor].main,
            },
        },
    };
}
function SuccessIcon() {
    return (React__default["default"].createElement(SvgIcon, { viewBox: "0 0 24 24", fontSize: "inherit" },
        React__default["default"].createElement("path", { fillRule: "evenodd", clipRule: "evenodd", d: "M12 2.25C10.7196 2.25 9.45176 2.50219 8.26884 2.99217C7.08591 3.48216 6.01108 4.20034 5.10571 5.10571C4.20034 6.01108 3.48216 7.08591 2.99217 8.26884C2.50219 9.45176 2.25 10.7196 2.25 12C2.25 13.2804 2.50219 14.5482 2.99217 15.7312C3.48216 16.9141 4.20034 17.9889 5.10571 18.8943C6.01108 19.7997 7.08591 20.5178 8.26884 21.0078C9.45176 21.4978 10.7196 21.75 12 21.75C13.2804 21.75 14.5482 21.4978 15.7312 21.0078C16.9141 20.5178 17.9889 19.7997 18.8943 18.8943C19.7997 17.9889 20.5178 16.9141 21.0078 15.7312C21.4978 14.5482 21.75 13.2804 21.75 12C21.75 10.7196 21.4978 9.45176 21.0078 8.26884C20.5178 7.08591 19.7997 6.01108 18.8943 5.10571C17.9889 4.20034 16.9141 3.48216 15.7312 2.99217C14.5482 2.50219 13.2804 2.25 12 2.25ZM16.2415 10.0563C16.5344 9.76339 16.5344 9.28852 16.2415 8.99563C15.9486 8.70273 15.4737 8.70273 15.1809 8.99563L10.7631 13.4134L8.81939 11.4697C8.5265 11.1768 8.05163 11.1768 7.75873 11.4697C7.46584 11.7626 7.46584 12.2374 7.75873 12.5303L10.2328 15.0044C10.3734 15.145 10.5642 15.224 10.7631 15.224C10.962 15.224 11.1528 15.145 11.2934 15.0044L16.2415 10.0563Z" })));
}
function InfoIcon() {
    return (React__default["default"].createElement(SvgIcon, { viewBox: "0 0 24 24", fontSize: "inherit" },
        React__default["default"].createElement("path", { fillRule: "evenodd", clipRule: "evenodd", d: "M2.25 11.75C2.25 6.36522 6.61522 2 12 2C17.3848 2 21.75 6.36522 21.75 11.75C21.75 17.1348 17.3848 21.5 12 21.5C6.61522 21.5 2.25 17.1348 2.25 11.75ZM11.25 7.75C11.25 7.33579 11.5858 7 12 7H12.01C12.4242 7 12.76 7.33579 12.76 7.75C12.76 8.16421 12.4242 8.5 12.01 8.5H12C11.5858 8.5 11.25 8.16421 11.25 7.75ZM10.25 11.75C10.25 11.3358 10.5858 11 11 11H12C12.4142 11 12.75 11.3358 12.75 11.75V15H13C13.4142 15 13.75 15.3358 13.75 15.75C13.75 16.1642 13.4142 16.5 13 16.5H12C11.5858 16.5 11.25 16.1642 11.25 15.75V12.5H11C10.5858 12.5 10.25 12.1642 10.25 11.75Z" })));
}
function ErrorIcon() {
    return (React__default["default"].createElement(SvgIcon, { viewBox: "0 0 24 24", fontSize: "inherit" },
        React__default["default"].createElement("path", { fillRule: "evenodd", clipRule: "evenodd", d: "M8.7 2.25C8.46249 2.25 8.23103 2.29047 8.0079 2.38964C7.78802 2.48736 7.61395 2.62539 7.46967 2.76967L2.76967 7.46967C2.62539 7.61395 2.48736 7.78802 2.38964 8.0079C2.29047 8.23103 2.25 8.46249 2.25 8.7V15.3C2.25 15.5375 2.29047 15.769 2.38964 15.9921C2.48736 16.212 2.62539 16.3861 2.76967 16.5303L7.46967 21.2303C7.61395 21.3746 7.78802 21.5126 8.0079 21.6104C8.23103 21.7095 8.46249 21.75 8.7 21.75H15.3C15.5375 21.75 15.769 21.7095 15.9921 21.6104C16.212 21.5126 16.3861 21.3746 16.5303 21.2303L21.2303 16.5303C21.3746 16.3861 21.5126 16.212 21.6104 15.9921C21.7095 15.769 21.75 15.5375 21.75 15.3V8.7C21.75 8.46249 21.7095 8.23103 21.6104 8.0079C21.5126 7.78802 21.3746 7.61395 21.2303 7.46967L16.5303 2.76967C16.3861 2.62539 16.212 2.48736 15.9921 2.38964C15.769 2.29047 15.5375 2.25 15.3 2.25H8.7ZM12.75 8C12.75 7.58579 12.4142 7.25 12 7.25C11.5858 7.25 11.25 7.58579 11.25 8V12C11.25 12.4142 11.5858 12.75 12 12.75C12.4142 12.75 12.75 12.4142 12.75 12V8ZM12 15.25C11.5858 15.25 11.25 15.5858 11.25 16C11.25 16.4142 11.5858 16.75 12 16.75H12.01C12.4242 16.75 12.76 16.4142 12.76 16C12.76 15.5858 12.4242 15.25 12.01 15.25H12Z" })));
}
function WarningIcon() {
    return (React__default["default"].createElement(SvgIcon, { viewBox: "0 0 24 24", fontSize: "inherit" },
        React__default["default"].createElement("path", { fillRule: "evenodd", clipRule: "evenodd", d: "M11.9932 3C11.5018 3 11.0194 3.13134 10.596 3.38038C10.175 3.62805 9.82781 3.98314 9.59 4.40906L2.4909 16.6309C2.47341 16.661 2.45804 16.6923 2.44491 16.7246C2.27977 17.1303 2.21428 17.5695 2.25392 18.0056C2.29356 18.4416 2.43717 18.8619 2.67276 19.2313C2.90835 19.6008 3.22909 19.9086 3.6082 20.1291C3.98731 20.3496 4.41379 20.4764 4.85202 20.499C4.88374 20.5006 4.9151 20.5003 4.94598 20.498C4.96405 20.4993 4.98229 20.5 5.00069 20.5H19.0057L19.011 20.5C19.4598 20.4968 19.9011 20.3841 20.2962 20.1718C20.6914 19.9594 21.0285 19.6537 21.2781 19.2815C21.5277 18.9093 21.6822 18.4818 21.7282 18.0362C21.7742 17.5907 21.7102 17.1408 21.5419 16.7256C21.5287 16.693 21.5132 16.6613 21.4955 16.6309L14.3964 4.40904C14.1586 3.98312 13.8114 3.62805 13.3904 3.38038C12.9671 3.13134 12.4846 3 11.9932 3ZM12.7538 8.76945C12.7538 8.35599 12.4179 8.02081 12.0035 8.02081C11.5891 8.02081 11.2532 8.35599 11.2532 8.76945V12.7658C11.2532 13.1793 11.5891 13.5145 12.0035 13.5145C12.4179 13.5145 12.7538 13.1793 12.7538 12.7658V8.76945ZM12.7538 15.7586C12.7538 15.3451 12.4179 15.0099 12.0035 15.0099C11.5891 15.0099 11.2532 15.3451 11.2532 15.7586V15.7686C11.2532 16.182 11.5891 16.5172 12.0035 16.5172C12.4179 16.5172 12.7538 16.182 12.7538 15.7686V15.7586Z" })));
}

const hardCodedColor = {
    primary: '#C00BB9',
    global: '#17929B',
};
const getLinkTextColor = (color = 'primary', variant = 'text') => {
    if (!color) {
        return;
    }
    if (color === 'inherit') {
        return 'inherit';
    }
    if (variant === 'contained') {
        return `${color}.contrastText`;
    }
    if (color in hardCodedColor) {
        return hardCodedColor[color];
    }
    return `${color}.main`;
};
// Modified reason: Material UI Typescript problem with the 'component' prop and handling the link global style overrides.
const Button = React__default["default"].forwardRef((props, ref) => {
    const { sx = {}, ...rootProps } = props;
    let linkSx = {};
    // Temporary solution to prevent global styles from affecting the link color.
    if (rootProps.href) {
        linkSx = {
            [LINK_PSEUDO_SELECTORS]: {
                color: getLinkTextColor(rootProps.color, rootProps.variant),
            },
        };
    }
    return (React__default["default"].createElement(MuiButton__default["default"], { ...props, sx: {
            ...linkSx,
            ...sx,
        }, ref: ref }));
});

const { slots: slots$3, classNames: classNames$3 } = createSlots('AlertAction', ['root']);
const Root$3 = styles.styled(Button, slots$3.root)({});
const AlertAction = React__default["default"].forwardRef((inProps, ref) => {
    const props = styles.useThemeProps({ props: inProps, name: slots$3.root.name });
    return (React__default["default"].createElement(Root$3, { ...props, size: "small", ref: ref, className: clsx__default["default"]([[classNames$3.root, props.className]]), ownerState: props }));
});
AlertAction.defaultProps = {
    color: 'inherit',
    variant: 'outlined',
};

const AlertTitle = React__default["default"].forwardRef((props, ref) => {
    return (React__default["default"].createElement(MuiAlertTitle__default["default"], { ...props, ref: ref }));
});

const AppBar = React__default["default"].forwardRef((props, ref) => {
    return (React__default["default"].createElement(MuiAppBar__default["default"], { ...props, ref: ref }));
});

const Autocomplete = React__default["default"].forwardRef((props, ref) => {
    return (React__default["default"].createElement(MuiAutocomplete__default["default"], { ...props, ref: ref }));
});

const Avatar = React__default["default"].forwardRef((props, ref) => {
    return (React__default["default"].createElement(MuiAvatar__default["default"], { ...props, ref: ref }));
});

const AvatarGroup = React__default["default"].forwardRef((props, ref) => {
    return (React__default["default"].createElement(MuiAvatarGroup__default["default"], { ...props, ref: ref }));
});

const Backdrop = React__default["default"].forwardRef((props, ref) => {
    return (React__default["default"].createElement(MuiBackdrop__default["default"], { ...props, ref: ref }));
});

const Badge = React__default["default"].forwardRef((props, ref) => {
    return (React__default["default"].createElement(MuiBadge__default["default"], { ...props, ref: ref }));
});

const BottomNavigation = React__default["default"].forwardRef((props, ref) => {
    return (React__default["default"].createElement(MuiBottomNavigation__default["default"], { ...props, ref: ref }));
});

const BottomNavigationAction = React__default["default"].forwardRef((props, ref) => {
    return (React__default["default"].createElement(MuiBottomNavigationAction__default["default"], { ...props, ref: ref }));
});

const Box = React__default["default"].forwardRef((props, ref) => {
    return (React__default["default"].createElement(MuiBox__default["default"], { ...props, ref: ref }));
});

const Breadcrumbs = React__default["default"].forwardRef((props, ref) => {
    return (React__default["default"].createElement(MuiBreadcrumbs__default["default"], { ...props, ref: ref }));
});

const ButtonBase = React__default["default"].forwardRef((props, ref) => {
    return (React__default["default"].createElement(MuiButtonBase__default["default"], { ...props, ref: ref }));
});

const ButtonGroup = React__default["default"].forwardRef((props, ref) => {
    return (React__default["default"].createElement(MuiButtonGroup__default["default"], { ...props, ref: ref }));
});

const Card = React__default["default"].forwardRef((props, ref) => {
    return (React__default["default"].createElement(MuiCard__default["default"], { ...props, ref: ref }));
});

const CardActionArea = React__default["default"].forwardRef((props, ref) => {
    return (React__default["default"].createElement(MuiCardActionArea__default["default"], { ...props, ref: ref }));
});

const CardActions = React__default["default"].forwardRef((props, ref) => {
    return (React__default["default"].createElement(MuiCardActions__default["default"], { ...props, ref: ref }));
});

const CardContent = React__default["default"].forwardRef((props, ref) => {
    return (React__default["default"].createElement(MuiCardContent__default["default"], { ...props, ref: ref }));
});

const CardHeader = React__default["default"].forwardRef((props, ref) => {
    return (React__default["default"].createElement(MuiCardHeader__default["default"], { ...props, ref: ref }));
});

const CardMedia = React__default["default"].forwardRef((props, ref) => {
    return (React__default["default"].createElement(MuiCardMedia__default["default"], { ...props, ref: ref }));
});

const Checkbox = React__default["default"].forwardRef((props, ref) => {
    return (React__default["default"].createElement(MuiCheckbox__default["default"], { ...props, ref: ref }));
});

const Chip = React__default["default"].forwardRef((props, ref) => {
    return (React__default["default"].createElement(MuiChip__default["default"], { ...props, ref: ref }));
});

const CircularProgress = React__default["default"].forwardRef((props, ref) => {
    return (React__default["default"].createElement(MuiCircularProgress__default["default"], { ...props, ref: ref }));
});

const ClickAwayListener = React__default["default"].forwardRef((props, ref) => {
    return (React__default["default"].createElement(MuiClickAwayListener__default["default"], { ...props, ref: ref }));
});

const Collapse = React__default["default"].forwardRef((props, ref) => {
    return (React__default["default"].createElement(MuiCollapse__default["default"], { ...props, ref: ref }));
});

const Container = React__default["default"].forwardRef((props, ref) => {
    return (React__default["default"].createElement(MuiContainer__default["default"], { ...props, ref: ref }));
});

const CssBaseline = (props) => (React__default["default"].createElement(MuiCssBaseline__default["default"], { ...props }));

const InputAdornment = React__default["default"].forwardRef((props, ref) => {
    return (React__default["default"].createElement(MuiInputAdornment__default["default"], { ...props, ref: ref }));
});

const TextField = React__default["default"].forwardRef((props, ref) => {
    return (React__default["default"].createElement(MuiTextField__default["default"], { ...props, ref: ref }));
});

const CalendarIcon = React__default["default"].forwardRef((props, ref) => {
    return (React__default["default"].createElement(SvgIcon, { viewBox: "0 0 24 24", ...props, ref: ref },
        React__default["default"].createElement("path", { fillRule: "evenodd", clipRule: "evenodd", d: "M8 2.25C8.41421 2.25 8.75 2.58579 8.75 3V4.25H15.25V3C15.25 2.58579 15.5858 2.25 16 2.25C16.4142 2.25 16.75 2.58579 16.75 3V4.25H18C19.5188 4.25 20.75 5.48122 20.75 7V19C20.75 20.5188 19.5188 21.75 18 21.75H6C4.48122 21.75 3.25 20.5188 3.25 19V7C3.25 5.48122 4.48122 4.25 6 4.25H7.25V3C7.25 2.58579 7.58579 2.25 8 2.25ZM7.25 5.75H6C5.30964 5.75 4.75 6.30964 4.75 7V10.25H19.25V7C19.25 6.30964 18.6904 5.75 18 5.75H16.75V7C16.75 7.41421 16.4142 7.75 16 7.75C15.5858 7.75 15.25 7.41421 15.25 7V5.75H8.75V7C8.75 7.41421 8.41421 7.75 8 7.75C7.58579 7.75 7.25 7.41421 7.25 7V5.75ZM19.25 11.75H4.75V19C4.75 19.6904 5.30964 20.25 6 20.25H18C18.6904 20.25 19.25 19.6904 19.25 19V11.75Z" })));
});
const DatePicker = React__default["default"].forwardRef(({ slots = {}, slotProps = {}, ...props }, ref) => {
    return (React__default["default"].createElement(DatePicker$1.DatePicker, { ...props, ref: ref, slots: {
            clearButton: IconButton,
            inputAdornment: InputAdornment,
            nextIconButton: IconButton,
            openPickerButton: IconButton,
            openPickerIcon: CalendarIcon,
            previousIconButton: IconButton,
            switchViewButton: IconButton,
            textField: TextField,
            ...slots,
        }, slotProps: {
            layout: {
                sx: {
                    '& .MuiDayCalendar-root': {
                        // The calendar layout should always be aligned to the left.
                        direction: 'ltr /* @noflip */',
                    },
                    '& .MuiYearCalendar-root': {
                        // The year selection layout should always be aligned to the left.
                        direction: 'ltr /* @noflip */',
                    },
                },
            },
            ...slotProps,
        } }));
});

// Modified reason: Material UI Typescript problem with the 'component' prop.
const Dialog = React__default["default"].forwardRef((props, ref) => {
    return (React__default["default"].createElement(MuiDialog__default["default"], { ...props, ref: ref }));
});

const DialogActions = React__default["default"].forwardRef((props, ref) => {
    return (React__default["default"].createElement(MuiDialogActions__default["default"], { ...props, ref: ref }));
});

const DialogContent = React__default["default"].forwardRef((props, ref) => {
    return (React__default["default"].createElement(MuiDialogContent__default["default"], { ...props, ref: ref }));
});

const DialogContentText = React__default["default"].forwardRef((props, ref) => {
    return (React__default["default"].createElement(MuiDialogContentText__default["default"], { ...props, ref: ref }));
});

const Toolbar = React__default["default"].forwardRef((props, ref) => {
    return (React__default["default"].createElement(MuiToolbar__default["default"], { ...props, ref: ref }));
});

const Stack = React__default["default"].forwardRef((props, ref) => {
    return (React__default["default"].createElement(MuiStack__default["default"], { ...props, ref: ref }));
});

const ElementorLogo = (props) => {
    return (React__default["default"].createElement(SvgIcon, { viewBox: "0 0 32 32", ...props },
        React__default["default"].createElement("path", { fillRule: "evenodd", clipRule: "evenodd", d: "M2.69648 24.8891C0.938383 22.2579 0 19.1645 0 16C0 11.7566 1.68571 7.68687 4.68629 4.68629C7.68687 1.68571 11.7566 0 16 0C19.1645 0 22.2579 0.938383 24.8891 2.69648C27.5203 4.45459 29.5711 6.95344 30.7821 9.87706C31.9931 12.8007 32.3099 16.0177 31.6926 19.1214C31.0752 22.2251 29.5514 25.0761 27.3137 27.3137C25.0761 29.5514 22.2251 31.0752 19.1214 31.6926C16.0177 32.3099 12.8007 31.9931 9.87706 30.7821C6.95344 29.5711 4.45459 27.5203 2.69648 24.8891ZM12.0006 9.33281H9.33437V22.6665H12.0006V9.33281ZM22.6657 9.33281H14.6669V11.9991H22.6657V9.33281ZM22.6657 14.6654H14.6669V17.3316H22.6657V14.6654ZM22.6657 20.0003H14.6669V22.6665H22.6657V20.0003Z" })));
};
const StyledElementorLogo = styles.styled(ElementorLogo)(({ theme }) => ({
    width: theme.spacing(3),
    height: theme.spacing(3),
    '& path': {
        fill: theme.palette.text.primary,
    },
    marginRight: theme.spacing(1),
}));
const StyledCustomLogo = styles.styled('span')(({ theme }) => ({
    marginRight: theme.spacing(1),
}));
const Logo = ({ logo, ...rest }) => {
    if (logo === false) {
        return null;
    }
    if (logo) {
        return React__default["default"].createElement(StyledCustomLogo, null, logo);
    }
    return React__default["default"].createElement(StyledElementorLogo, { ...rest });
};
const { slots: slots$2, classNames: classNames$2 } = createSlots('DialogHeader', ['root', 'logo', 'toolbar']);
const Root$2 = styles.styled(AppBar, slots$2.root)({
    '& .MuiDialogTitle-root': {
        padding: 0,
    },
});
const ToolbarSlot = styles.styled(Toolbar, slots$2.toolbar)({});
const DialogHeader = React__default["default"].forwardRef((inProps, ref) => {
    const props = styles.useThemeProps({ props: inProps, name: slots$2.root.name });
    const { slotProps = {}, logo, onClose, ...rootProps } = props;
    return (React__default["default"].createElement(Root$2, { ...rootProps, ref: ref, className: clsx__default["default"]([[classNames$2.root, rootProps.className]]), ownerState: props },
        React__default["default"].createElement(ToolbarSlot, { variant: "dense", ...slotProps.toolbar, className: clsx__default["default"]([classNames$2.toolbar, slotProps.toolbar?.className]), ownerState: props },
            React__default["default"].createElement(Logo, { logo: logo, className: clsx__default["default"]([classNames$2.logo, slotProps.logo?.className]) }),
            React__default["default"].createElement(Stack, { direction: "row", alignItems: "center", flex: 1 }, props.children),
            onClose && (React__default["default"].createElement(CloseButton, { edge: "end", onClick: onClose, sx: { '&.MuiButtonBase-root': { ml: 0.5 } } })))));
});
DialogHeader.defaultProps = {
    color: 'transparent',
    position: 'relative',
};

const { slots: slots$1, classNames: classNames$1 } = createSlots('DialogHeaderGroup', ['root']);
const Root$1 = styles.styled(Box, slots$1.root)(({ theme, ownerState }) => {
    const { disableSpacing, disableGutters, gutterLeftAuto, gutterRightAuto } = ownerState;
    return {
        display: 'flex',
        alignItems: 'center',
        gap: disableSpacing ? undefined : theme.spacing(1),
        '.MuiDialogHeaderGroup-root + &.MuiDialogHeaderGroup-root': (disableGutters || gutterLeftAuto) ? undefined : {
            marginLeft: theme.spacing(2),
        },
        marginLeft: gutterLeftAuto ? 'auto' : undefined,
        marginRight: gutterRightAuto ? 'auto' : undefined,
    };
});
const DialogHeaderGroup = React__default["default"].forwardRef((inProps, ref) => {
    const props = styles.useThemeProps({ props: inProps, name: slots$1.root.name });
    const { disableSpacing, disableGutters, gutterLeftAuto, gutterRightAuto, ...rootProps } = props;
    const ownerState = {
        disableSpacing,
        disableGutters,
        gutterLeftAuto,
        gutterRightAuto,
    };
    return (React__default["default"].createElement(Root$1, { ...rootProps, ref: ref, className: clsx__default["default"]([[classNames$1.root, rootProps.className]]), ownerState: ownerState }));
});

// Modified reason: modifying the default props.
const DialogTitle = React__default["default"].forwardRef((props, ref) => {
    return (React__default["default"].createElement(MuiDialogTitle__default["default"], { ...props, ref: ref }));
});
DialogTitle.defaultProps = {
    variant: 'subtitle1',
};

const Divider = React__default["default"].forwardRef((props, ref) => {
    return (React__default["default"].createElement(MuiDivider__default["default"], { ...props, ref: ref }));
});

// Modified reason: Material UI Typescript problem with the 'component' prop.
const Drawer = React__default["default"].forwardRef((props, ref) => {
    return (React__default["default"].createElement(MuiDrawer__default["default"], { ...props, ref: ref }));
});

const Fab = React__default["default"].forwardRef((props, ref) => {
    return (React__default["default"].createElement(MuiFab__default["default"], { ...props, ref: ref }));
});

const Fade = React__default["default"].forwardRef((props, ref) => {
    return (React__default["default"].createElement(MuiFade__default["default"], { ...props, ref: ref }));
});

const FilledInput = React__default["default"].forwardRef((props, ref) => {
    return (React__default["default"].createElement(MuiFilledInput__default["default"], { ...props, ref: ref }));
});

const FormControl = React__default["default"].forwardRef((props, ref) => {
    return (React__default["default"].createElement(MuiFormControl__default["default"], { ...props, ref: ref }));
});

const FormControlLabel = React__default["default"].forwardRef((props, ref) => {
    return (React__default["default"].createElement(MuiFormControlLabel__default["default"], { ...props, ref: ref }));
});

const FormGroup = React__default["default"].forwardRef((props, ref) => {
    return (React__default["default"].createElement(MuiFormGroup__default["default"], { ...props, ref: ref }));
});

const FormHelperText = React__default["default"].forwardRef((props, ref) => {
    return (React__default["default"].createElement(MuiFormHelperText__default["default"], { ...props, ref: ref }));
});

const FormLabel = React__default["default"].forwardRef((props, ref) => {
    return (React__default["default"].createElement(MuiFormLabel__default["default"], { ...props, ref: ref }));
});

const Grid = React__default["default"].forwardRef((props, ref) => {
    return (React__default["default"].createElement(MuiGrid__default["default"], { ...props, ref: ref }));
});

const Grow = React__default["default"].forwardRef((props, ref) => {
    return (React__default["default"].createElement(MuiGrow__default["default"], { ...props, ref: ref }));
});

const Icon = React__default["default"].forwardRef((props, ref) => {
    return (React__default["default"].createElement(MuiIcon__default["default"], { ...props, ref: ref }));
});

const ImageList = React__default["default"].forwardRef((props, ref) => {
    return (React__default["default"].createElement(MuiImageList__default["default"], { ...props, ref: ref }));
});

const ImageListItem = React__default["default"].forwardRef((props, ref) => {
    return (React__default["default"].createElement(MuiImageListItem__default["default"], { ...props, ref: ref }));
});

const ImageListItemBar = React__default["default"].forwardRef((props, ref) => {
    return (React__default["default"].createElement(MuiImageListItemBar__default["default"], { ...props, ref: ref }));
});

const Input = React__default["default"].forwardRef((props, ref) => {
    return (React__default["default"].createElement(MuiInput__default["default"], { ...props, ref: ref }));
});

const InputBase = React__default["default"].forwardRef((props, ref) => {
    return (React__default["default"].createElement(MuiInputBase__default["default"], { ...props, ref: ref }));
});

const InputLabel = React__default["default"].forwardRef((props, ref) => {
    return (React__default["default"].createElement(MuiInputLabel__default["default"], { ...props, ref: ref }));
});

const LinearProgress = React__default["default"].forwardRef((props, ref) => {
    return (React__default["default"].createElement(MuiLinearProgress__default["default"], { ...props, ref: ref }));
});

/**
 * All supported colors for the Link component that can be passed as a string (without using primary.main, secondary.main, etc.).
 *
 * @see https://github.com/mui/material-ui/blob/f3945c1227afc13fbf732e66875095a793af2f10/packages/mui-material/src/Typography/Typography.js#L82
 */
const colorTransformations = {
    primary: 'primary.main',
    textPrimary: 'text.primary',
    secondary: 'secondary.main',
    textSecondary: 'text.secondary',
    error: 'error.main',
};
// The sx prop does not support the color values of the Link component, so we need to transform them to the correct format.
const convertToSxSupportedColor = (color) => {
    // Temporary customization until the pink color will be replaced with purple.
    if (color === 'primary.main' || color === 'primary') {
        return '#C00BB9';
    }
    if (color === 'global.main') {
        return '#17929B';
    }
    return colorTransformations[color] || color;
};
// Modified reason: the Link component does not provide default style for :hover, therefore global CSS can override the link color.
const Link = React__default["default"].forwardRef((inProps, ref) => {
    const { sx = {}, ...props } = inProps;
    const sxSupportedColor = convertToSxSupportedColor(props.color);
    return (React__default["default"].createElement(MuiLink__default["default"], { ...props, color: sxSupportedColor, sx: {
            // Setting the pseudo-classes color by default to prevent global CSS from overriding the link color on :hover.
            [LINK_PSEUDO_SELECTORS]: {
                color: sxSupportedColor,
            },
            ...sx,
        }, ref: ref }));
});
Link.defaultProps = {
    color: 'primary.main',
};

// Modified reason: Material UI Typescript problem with the 'component' prop.
const List = React__default["default"].forwardRef((props, ref) => {
    return (React__default["default"].createElement(MuiList__default["default"], { ...props, ref: ref }));
});

// Modified reason: Material UI Typescript problem with the 'component' prop.
const ListItem = React__default["default"].forwardRef((props, ref) => {
    return (React__default["default"].createElement(MuiListItem__default["default"], { ...props, ref: ref }));
});

const ListItemAvatar = React__default["default"].forwardRef((props, ref) => {
    return (React__default["default"].createElement(MuiListItemAvatar__default["default"], { ...props, ref: ref }));
});

// Modified reason: Material UI Typescript problem with the 'component' prop.
const ListItemButton = React__default["default"].forwardRef((props, ref) => {
    return (React__default["default"].createElement(MuiListItemButton__default["default"], { ...props, ref: ref }));
});

const ListItemIcon = React__default["default"].forwardRef((props, ref) => {
    return (React__default["default"].createElement(MuiListItemIcon__default["default"], { ...props, ref: ref }));
});

const ListItemSecondaryAction = React__default["default"].forwardRef((props, ref) => {
    return (React__default["default"].createElement(MuiListItemSecondaryAction__default["default"], { ...props, ref: ref }));
});

const ListItemText = React__default["default"].forwardRef((props, ref) => {
    return (React__default["default"].createElement(MuiListItemText__default["default"], { ...props, ref: ref }));
});

const ListSubheader = React__default["default"].forwardRef((props, ref) => {
    return (React__default["default"].createElement(MuiListSubheader__default["default"], { ...props, ref: ref }));
});

// TODO: fix the any type, only when passing all props there is a problem with the type, when using only props.children it works fine.
// The need for passing the props is to allow overriding the default dateAdapter, when wanting to use a different dates library.
const LocalizationProvider = (props) => (
// eslint-disable-next-line @typescript-eslint/no-explicit-any
React__default["default"].createElement(LocalizationProvider$1.LocalizationProvider, { dateAdapter: AdapterDayjs.AdapterDayjs, ...props }));

// Modified reason: Material UI Typescript problem with the 'component' prop.
const Menu = React__default["default"].forwardRef((props, ref) => {
    return (React__default["default"].createElement(MuiMenu__default["default"], { ...props, ref: ref }));
});

// Modified reason: Material UI Typescript problem with the 'component' prop.
const MenuItem = React__default["default"].forwardRef((props, ref) => {
    return (React__default["default"].createElement(MuiMenuItem__default["default"], { ...props, ref: ref }));
});

// Modified reason: Material UI Typescript problem with the 'component' prop.
const MenuList = React__default["default"].forwardRef((props, ref) => {
    return (React__default["default"].createElement(MuiMenuList__default["default"], { ...props, ref: ref }));
});

const MobileStepper = React__default["default"].forwardRef((props, ref) => {
    return (React__default["default"].createElement(MuiMobileStepper__default["default"], { ...props, ref: ref }));
});

const Modal = React__default["default"].forwardRef((props, ref) => {
    return (React__default["default"].createElement(MuiModal__default["default"], { ...props, ref: ref }));
});

const NativeSelect = React__default["default"].forwardRef((props, ref) => {
    return (React__default["default"].createElement(MuiNativeSelect__default["default"], { ...props, ref: ref }));
});

const OutlinedInput = React__default["default"].forwardRef((props, ref) => {
    return (React__default["default"].createElement(MuiOutlinedInput__default["default"], { ...props, ref: ref }));
});

const Pagination = React__default["default"].forwardRef((props, ref) => {
    return (React__default["default"].createElement(MuiPagination__default["default"], { ...props, ref: ref }));
});

const PaginationItem = React__default["default"].forwardRef((props, ref) => {
    return (React__default["default"].createElement(MuiPaginationItem__default["default"], { ...props, ref: ref }));
});

// Modified reason: Material UI Typescript problem with the 'component' prop.
const Paper = React__default["default"].forwardRef((props, ref) => {
    return (React__default["default"].createElement(MuiPaper__default["default"], { ...props, ref: ref }));
});

const Popover = React__default["default"].forwardRef((props, ref) => {
    return (React__default["default"].createElement(MuiPopover__default["default"], { ...props, ref: ref }));
});

// Modified reason: the placeholder prop is mandatory, we need to verify it after @mui/material 5.15.12.
const Popper = React__default["default"].forwardRef((props, ref) => {
    return (React__default["default"].createElement(MuiPopper__default["default"], { placeholder: null, ...props, ref: ref }));
});

const Portal = (props) => (React__default["default"].createElement(MuiPortal__default["default"], { ...props }));

const Radio = React__default["default"].forwardRef((props, ref) => {
    return (React__default["default"].createElement(MuiRadio__default["default"], { ...props, ref: ref }));
});

const RadioGroup = React__default["default"].forwardRef((props, ref) => {
    return (React__default["default"].createElement(MuiRadioGroup__default["default"], { ...props, ref: ref }));
});

const Rating = React__default["default"].forwardRef((props, ref) => {
    return (React__default["default"].createElement(MuiRating__default["default"], { ...props, ref: ref }));
});

const ScopedCssBaseline = (props) => (React__default["default"].createElement(MuiScopedCssBaseline__default["default"], { ...props }));

const Select = React__default["default"].forwardRef((props, ref) => {
    return (React__default["default"].createElement(MuiSelect__default["default"], { ...props, ref: ref }));
});

const Skeleton = React__default["default"].forwardRef((props, ref) => {
    return (React__default["default"].createElement(MuiSkeleton__default["default"], { ...props, ref: ref }));
});

const Slide = React__default["default"].forwardRef((props, ref) => {
    return (React__default["default"].createElement(MuiSlide__default["default"], { ...props, ref: ref }));
});

const Slider = React__default["default"].forwardRef((props, ref) => {
    return (React__default["default"].createElement(MuiSlider__default["default"], { ...props, ref: ref }));
});

const Snackbar = React__default["default"].forwardRef((props, ref) => {
    return (React__default["default"].createElement(MuiSnackbar__default["default"], { ...props, ref: ref }));
});

const SnackbarContent = React__default["default"].forwardRef((props, ref) => {
    return (React__default["default"].createElement(MuiSnackbarContent__default["default"], { ...props, ref: ref }));
});

const SpeedDial = React__default["default"].forwardRef((props, ref) => {
    return (React__default["default"].createElement(MuiSpeedDial__default["default"], { ...props, ref: ref }));
});

const SpeedDialAction = React__default["default"].forwardRef((props, ref) => {
    return (React__default["default"].createElement(MuiSpeedDialAction__default["default"], { ...props, ref: ref }));
});

const SpeedDialIcon = React__default["default"].forwardRef((props, ref) => {
    return (React__default["default"].createElement(MuiSpeedDialIcon__default["default"], { ...props, ref: ref }));
});

// TODO: replace to icon from a dedicated package once available.
const ArrowDropDownIcon = () => {
    return (React__default["default"].createElement(SvgIcon, { viewBox: "0 0 24 24", sx: { fill: '#fff' } },
        React__default["default"].createElement("path", { fillRule: "evenodd", clipRule: "evenodd", d: "M17.2929 9.29289C17.6834 8.90237 18.3166 8.90237 18.7071 9.29289C19.0976 9.68342 19.0976 10.3166 18.7071 10.7071L12.7071 16.7071C12.3166 17.0976 11.6834 17.0976 11.2929 16.7071L5.29289 10.7071C4.90237 10.3166 4.90237 9.68342 5.29289 9.29289C5.68342 8.90237 6.31658 8.90237 6.70711 9.29289L12 14.5858L17.2929 9.29289Z" })));
};
const { slots, classNames } = createSlots('SplitButton', ['root', 'mainButton', 'caretButton']);
const Root = styles.styled(ButtonGroup, slots.root)({});
const MainButton = styles.styled(Button, slots.mainButton)({});
const CaretButton = styles.styled(Button, slots.caretButton)(() => ({ paddingInline: 0 }));
const SplitButton = React__default["default"].forwardRef((inProps, ref) => {
    const props = styles.useThemeProps({ props: inProps, name: slots.root.name });
    const { onClick, children, MainButtonProps = {}, CaretButtonProps = {}, slotProps = {}, ...rootProps } = props;
    return (React__default["default"].createElement(Root, { ...rootProps, ref: ref, className: clsx__default["default"]([[classNames.root, rootProps.className]]), ownerState: props },
        React__default["default"].createElement(MainButton, { onClick: onClick, ...MainButtonProps, ...slotProps.mainButton, className: clsx__default["default"]([classNames.mainButton, MainButtonProps.className, slotProps.mainButton?.className]), ownerState: props }, children),
        React__default["default"].createElement(CaretButton, { ...CaretButtonProps, ...slotProps.caretButton, className: clsx__default["default"]([classNames.caretButton, CaretButtonProps.className, slotProps.caretButton?.className]), ownerState: props })));
});
SplitButton.defaultProps = {
    variant: 'contained',
};
CaretButton.defaultProps = {
    children: React__default["default"].createElement(ArrowDropDownIcon, null),
};

const Step = React__default["default"].forwardRef((props, ref) => {
    return (React__default["default"].createElement(MuiStep__default["default"], { ...props, ref: ref }));
});

const StepButton = React__default["default"].forwardRef((props, ref) => {
    return (React__default["default"].createElement(MuiStepButton__default["default"], { ...props, ref: ref }));
});

const StepConnector = React__default["default"].forwardRef((props, ref) => {
    return (React__default["default"].createElement(MuiStepConnector__default["default"], { ...props, ref: ref }));
});

const StepContent = React__default["default"].forwardRef((props, ref) => {
    return (React__default["default"].createElement(MuiStepContent__default["default"], { ...props, ref: ref }));
});

const CheckedCircleIcon = React__default["default"].forwardRef((props, ref) => {
    const theme = styles.useTheme();
    return (React__default["default"].createElement(SvgIcon, { viewBox: "0 0 24 24", ...props, ref: ref },
        React__default["default"].createElement("g", { clipPath: "url(#clip0_2503_21505)" },
            React__default["default"].createElement("circle", { cx: "12", cy: "12", r: "12", fill: theme.palette.primary.main }),
            React__default["default"].createElement("path", { fill: theme.palette.primary.contrastText, fillRule: "evenodd", clipRule: "evenodd", d: "M17.7753 8.22472C17.5312 7.98064 17.1355 7.98064 16.8914 8.22472L10.6667 14.4494L7.77532 11.5581C7.53124 11.314 7.13551 11.314 6.89143 11.5581C6.64735 11.8021 6.64735 12.1979 6.89143 12.4419L10.2248 15.7753C10.4688 16.0194 10.8646 16.0194 11.1086 15.7753L17.7753 9.10861C18.0194 8.86453 18.0194 8.4688 17.7753 8.22472Z" })),
        React__default["default"].createElement("defs", null,
            React__default["default"].createElement("clipPath", { id: "clip0_2503_21505" },
                React__default["default"].createElement("rect", { width: "24", height: "24", rx: "12", fill: "white" })))));
});
// Modified reason: modifying the completed icon.
const StepIcon = React__default["default"].forwardRef((inProps, ref) => {
    const props = { ...inProps };
    if (props.completed) {
        // Overriding the completed icon using the 'as' prop because Mui does not have a dedicated prop for it.
        // see: https://github.com/mui/material-ui/blob/bcef13eef3208716bc33482dd5af3ac90ae18dc4/packages/mui-material/src/StepIcon/StepIcon.js#L87
        props.as = (svgProps) => React__default["default"].createElement(CheckedCircleIcon, { ...svgProps });
    }
    return (React__default["default"].createElement(MuiStepIcon__default["default"], { ...props, ref: ref }));
});

// Modified reason: modifying the default props and fixing the StepIconProps.label className.
const StepLabel = React__default["default"].forwardRef((inProps, ref) => {
    const props = { ...inProps };
    if (!props.componentsProps) {
        props.componentsProps = {};
    }
    if (!props.componentsProps.label) {
        props.componentsProps.label = {};
    }
    /**
     * There is a problem with how this component is built in Mui.
     * Currently when using this component outside of its context (without a Step and Stepper components as wrappers),
     * the component doesn't get the active and completed classNames, which define the required style for these states.
     */
    props.componentsProps = {
        ...props.componentsProps,
        label: {
            ...props.componentsProps.label,
            className: clsx__default["default"]([
                props.componentsProps.label.className,
                props.StepIconProps?.active && 'Mui-active',
                props.StepIconProps?.completed && 'Mui-completed',
                props.StepIconProps?.error && 'Mui-error',
            ]),
        },
    };
    return (React__default["default"].createElement(MuiStepLabel__default["default"], { ...props, ref: ref }));
});
StepLabel.defaultProps = {
    StepIconComponent: StepIcon,
};

const Stepper = React__default["default"].forwardRef((props, ref) => {
    return (React__default["default"].createElement(MuiStepper__default["default"], { ...props, ref: ref }));
});

const SwipeableDrawer = React__default["default"].forwardRef((props, ref) => {
    return (React__default["default"].createElement(MuiSwipeableDrawer__default["default"], { ...props, ref: ref }));
});

const Switch = React__default["default"].forwardRef((props, ref) => {
    return (React__default["default"].createElement(MuiSwitch__default["default"], { ...props, ref: ref }));
});

const Tab = React__default["default"].forwardRef((props, ref) => {
    return (React__default["default"].createElement(MuiTab__default["default"], { ...props, ref: ref }));
});

const TabScrollButton = React__default["default"].forwardRef((props, ref) => {
    return (React__default["default"].createElement(MuiTabScrollButton__default["default"], { ...props, ref: ref }));
});

const Table = React__default["default"].forwardRef((props, ref) => {
    return (React__default["default"].createElement(MuiTable__default["default"], { ...props, ref: ref }));
});

const TableBody = React__default["default"].forwardRef((props, ref) => {
    return (React__default["default"].createElement(MuiTableBody__default["default"], { ...props, ref: ref }));
});

const TableCell = React__default["default"].forwardRef((props, ref) => {
    return (React__default["default"].createElement(MuiTableCell__default["default"], { ...props, ref: ref }));
});

const TableContainer = React__default["default"].forwardRef((props, ref) => {
    return (React__default["default"].createElement(MuiTableContainer__default["default"], { ...props, ref: ref }));
});

const TableFooter = React__default["default"].forwardRef((props, ref) => {
    return (React__default["default"].createElement(MuiTableFooter__default["default"], { ...props, ref: ref }));
});

const TableHead = React__default["default"].forwardRef((props, ref) => {
    return (React__default["default"].createElement(MuiTableHead__default["default"], { ...props, ref: ref }));
});

const TablePagination = React__default["default"].forwardRef((props, ref) => {
    return (React__default["default"].createElement(MuiTablePagination__default["default"], { ...props, ref: ref }));
});

const TableRow = React__default["default"].forwardRef((props, ref) => {
    return (React__default["default"].createElement(MuiTableRow__default["default"], { ...props, ref: ref }));
});

const TableSortLabel = React__default["default"].forwardRef((props, ref) => {
    return (React__default["default"].createElement(MuiTableSortLabel__default["default"], { ...props, ref: ref }));
});

const Tabs = React__default["default"].forwardRef((props, ref) => {
    return (React__default["default"].createElement(MuiTabs__default["default"], { ...props, ref: ref }));
});

const TextareaAutosize = React__default["default"].forwardRef((props, ref) => {
    return (React__default["default"].createElement(MuiTextareaAutosize__default["default"], { ...props, ref: ref }));
});

const ClockIcon = React__default["default"].forwardRef((props, ref) => {
    return (React__default["default"].createElement(SvgIcon, { viewBox: "0 0 24 24", ...props, ref: ref },
        React__default["default"].createElement("path", { fillRule: "evenodd", clipRule: "evenodd", d: "M12 3.75C7.44365 3.75 3.75 7.44365 3.75 12C3.75 16.5563 7.44365 20.25 12 20.25C16.5563 20.25 20.25 16.5563 20.25 12C20.25 7.44365 16.5563 3.75 12 3.75ZM2.25 12C2.25 6.61522 6.61522 2.25 12 2.25C17.3848 2.25 21.75 6.61522 21.75 12C21.75 17.3848 17.3848 21.75 12 21.75C6.61522 21.75 2.25 17.3848 2.25 12ZM12 6.25C12.4142 6.25 12.75 6.58579 12.75 7V11.6893L15.5303 14.4697C15.8232 14.7626 15.8232 15.2374 15.5303 15.5303C15.2374 15.8232 14.7626 15.8232 14.4697 15.5303L11.4697 12.5303C11.329 12.3897 11.25 12.1989 11.25 12V7C11.25 6.58579 11.5858 6.25 12 6.25Z" })));
});
const TimePicker = React__default["default"].forwardRef(({ slots = {}, slotProps = {}, ...props }, ref) => {
    return (React__default["default"].createElement(TimePicker$1.TimePicker, { ...props, ref: ref, slots: {
            clearButton: IconButton,
            inputAdornment: InputAdornment,
            nextIconButton: IconButton,
            openPickerButton: IconButton,
            openPickerIcon: ClockIcon,
            previousIconButton: IconButton,
            textField: TextField,
            ...slots,
        }, slotProps: {
            layout: {
                sx: {
                    // The time controls should always be aligned to the left.
                    direction: 'ltr /* @noflip */',
                },
            },
            actionBar: {
                sx: {
                    // The action bar should be aligned according to the general direction (flipped by Material-UI in RTL mode).
                    direction: 'ltr',
                },
            },
            ...slotProps,
        } }));
});

// Modified reason: Material UI Typescript problem with the 'component' prop.
const ToggleButton = React__default["default"].forwardRef((props, ref) => {
    return (React__default["default"].createElement(MuiToggleButton__default["default"], { ...props, ref: ref }));
});

const ToggleButtonGroup = React__default["default"].forwardRef((props, ref) => {
    return (React__default["default"].createElement(MuiToggleButtonGroup__default["default"], { ...props, ref: ref }));
});

const Tooltip = React__default["default"].forwardRef((props, ref) => {
    return (React__default["default"].createElement(MuiTooltip__default["default"], { ...props, ref: ref }));
});

// Modified reason: Material UI Typescript problem with the 'component' prop.
const Typography = React__default["default"].forwardRef((props, ref) => {
    return (React__default["default"].createElement(MuiTypography__default["default"], { ...props, ref: ref }));
});

const Zoom = React__default["default"].forwardRef((props, ref) => {
    return (React__default["default"].createElement(MuiZoom__default["default"], { ...props, ref: ref }));
});

// The default Theme type in Mui is unknown, so we need to override it with our own Theme type.
const useMediaQuery = (useMediaQuery__default["default"]);

// eslint-disable-next-line @typescript-eslint/no-explicit-any
const withDirection = (Component) => {
    // eslint-disable-next-line @typescript-eslint/ban-ts-comment
    // @ts-ignore
    return styles.styled(Component)(({ theme }) => {
        return {
            // Flip the icon in RTL mode.
            transform: theme.direction === 'rtl' ? 'scaleX(-1)' : undefined,
        };
    });
};

Object.defineProperty(exports, 'usePopupState', {
	enumerable: true,
	get: function () { return hooks.usePopupState; }
});
exports.Accordion = Accordion;
exports.AccordionActions = AccordionActions;
exports.AccordionDetails = AccordionDetails;
exports.AccordionSummary = AccordionSummary;
exports.Alert = Alert;
exports.AlertAction = AlertAction;
exports.AlertTitle = AlertTitle;
exports.AppBar = AppBar;
exports.Autocomplete = Autocomplete;
exports.Avatar = Avatar;
exports.AvatarGroup = AvatarGroup;
exports.Backdrop = Backdrop;
exports.Badge = Badge;
exports.BottomNavigation = BottomNavigation;
exports.BottomNavigationAction = BottomNavigationAction;
exports.Box = Box;
exports.Breadcrumbs = Breadcrumbs;
exports.Button = Button;
exports.ButtonBase = ButtonBase;
exports.ButtonGroup = ButtonGroup;
exports.Card = Card;
exports.CardActionArea = CardActionArea;
exports.CardActions = CardActions;
exports.CardContent = CardContent;
exports.CardHeader = CardHeader;
exports.CardMedia = CardMedia;
exports.Checkbox = Checkbox;
exports.Chip = Chip;
exports.CircularProgress = CircularProgress;
exports.ClickAwayListener = ClickAwayListener;
exports.CloseButton = CloseButton;
exports.Collapse = Collapse;
exports.Container = Container;
exports.CssBaseline = CssBaseline;
exports.DatePicker = DatePicker;
exports.Dialog = Dialog;
exports.DialogActions = DialogActions;
exports.DialogContent = DialogContent;
exports.DialogContentText = DialogContentText;
exports.DialogHeader = DialogHeader;
exports.DialogHeaderGroup = DialogHeaderGroup;
exports.DialogTitle = DialogTitle;
exports.DirectionContext = DirectionContext;
exports.DirectionProvider = DirectionProvider;
exports.Divider = Divider;
exports.Drawer = Drawer;
exports.Fab = Fab;
exports.Fade = Fade;
exports.FilledInput = FilledInput;
exports.FormControl = FormControl;
exports.FormControlLabel = FormControlLabel;
exports.FormGroup = FormGroup;
exports.FormHelperText = FormHelperText;
exports.FormLabel = FormLabel;
exports.Grid = Grid;
exports.Grow = Grow;
exports.Icon = Icon;
exports.IconButton = IconButton;
exports.ImageList = ImageList;
exports.ImageListItem = ImageListItem;
exports.ImageListItemBar = ImageListItemBar;
exports.Input = Input;
exports.InputAdornment = InputAdornment;
exports.InputBase = InputBase;
exports.InputLabel = InputLabel;
exports.LinearProgress = LinearProgress;
exports.Link = Link;
exports.List = List;
exports.ListItem = ListItem;
exports.ListItemAvatar = ListItemAvatar;
exports.ListItemButton = ListItemButton;
exports.ListItemIcon = ListItemIcon;
exports.ListItemSecondaryAction = ListItemSecondaryAction;
exports.ListItemText = ListItemText;
exports.ListSubheader = ListSubheader;
exports.LocalizationProvider = LocalizationProvider;
exports.Menu = Menu;
exports.MenuItem = MenuItem;
exports.MenuList = MenuList;
exports.MobileStepper = MobileStepper;
exports.Modal = Modal;
exports.NativeSelect = NativeSelect;
exports.OutlinedInput = OutlinedInput;
exports.Pagination = Pagination;
exports.PaginationItem = PaginationItem;
exports.Paper = Paper;
exports.Popover = Popover;
exports.Popper = Popper;
exports.Portal = Portal;
exports.Radio = Radio;
exports.RadioGroup = RadioGroup;
exports.Rating = Rating;
exports.ScopedCssBaseline = ScopedCssBaseline;
exports.Select = Select;
exports.Skeleton = Skeleton;
exports.Slide = Slide;
exports.Slider = Slider;
exports.Snackbar = Snackbar;
exports.SnackbarContent = SnackbarContent;
exports.SpeedDial = SpeedDial;
exports.SpeedDialAction = SpeedDialAction;
exports.SpeedDialIcon = SpeedDialIcon;
exports.SplitButton = SplitButton;
exports.Stack = Stack;
exports.Step = Step;
exports.StepButton = StepButton;
exports.StepConnector = StepConnector;
exports.StepContent = StepContent;
exports.StepIcon = StepIcon;
exports.StepLabel = StepLabel;
exports.Stepper = Stepper;
exports.SvgIcon = SvgIcon;
exports.SwipeableDrawer = SwipeableDrawer;
exports.Switch = Switch;
exports.Tab = Tab;
exports.TabScrollButton = TabScrollButton;
exports.Table = Table;
exports.TableBody = TableBody;
exports.TableCell = TableCell;
exports.TableContainer = TableContainer;
exports.TableFooter = TableFooter;
exports.TableHead = TableHead;
exports.TablePagination = TablePagination;
exports.TableRow = TableRow;
exports.TableSortLabel = TableSortLabel;
exports.Tabs = Tabs;
exports.TextField = TextField;
exports.TextareaAutosize = TextareaAutosize;
exports.ThemeProvider = ThemeProvider;
exports.TimePicker = TimePicker;
exports.ToggleButton = ToggleButton;
exports.ToggleButtonGroup = ToggleButtonGroup;
exports.Toolbar = Toolbar;
exports.Tooltip = Tooltip;
exports.Typography = Typography;
exports.Zoom = Zoom;
exports.createSlots = createSlots;
exports.styles = index;
exports.useMediaQuery = useMediaQuery;
exports.withDirection = withDirection;
Object.keys(MuiAccordion$1).forEach(function (k) {
	if (k !== 'default' && !exports.hasOwnProperty(k)) Object.defineProperty(exports, k, {
		enumerable: true,
		get: function () { return MuiAccordion$1[k]; }
	});
});
Object.keys(MuiAccordionActions$1).forEach(function (k) {
	if (k !== 'default' && !exports.hasOwnProperty(k)) Object.defineProperty(exports, k, {
		enumerable: true,
		get: function () { return MuiAccordionActions$1[k]; }
	});
});
Object.keys(MuiAccordionDetails).forEach(function (k) {
	if (k !== 'default' && !exports.hasOwnProperty(k)) Object.defineProperty(exports, k, {
		enumerable: true,
		get: function () { return MuiAccordionDetails[k]; }
	});
});
Object.keys(MuiAccordionSummary$1).forEach(function (k) {
	if (k !== 'default' && !exports.hasOwnProperty(k)) Object.defineProperty(exports, k, {
		enumerable: true,
		get: function () { return MuiAccordionSummary$1[k]; }
	});
});
Object.keys(MuiSvgIcon).forEach(function (k) {
	if (k !== 'default' && !exports.hasOwnProperty(k)) Object.defineProperty(exports, k, {
		enumerable: true,
		get: function () { return MuiSvgIcon[k]; }
	});
});
Object.keys(MuiAlert).forEach(function (k) {
	if (k !== 'default' && !exports.hasOwnProperty(k)) Object.defineProperty(exports, k, {
		enumerable: true,
		get: function () { return MuiAlert[k]; }
	});
});
Object.keys(MuiIconButton$1).forEach(function (k) {
	if (k !== 'default' && !exports.hasOwnProperty(k)) Object.defineProperty(exports, k, {
		enumerable: true,
		get: function () { return MuiIconButton$1[k]; }
	});
});
Object.keys(styles).forEach(function (k) {
	if (k !== 'default' && !exports.hasOwnProperty(k)) Object.defineProperty(exports, k, {
		enumerable: true,
		get: function () { return styles[k]; }
	});
});
Object.keys(useMediaQuery$1).forEach(function (k) {
	if (k !== 'default' && !exports.hasOwnProperty(k)) Object.defineProperty(exports, k, {
		enumerable: true,
		get: function () { return useMediaQuery$1[k]; }
	});
});
Object.keys(MuiButton$1).forEach(function (k) {
	if (k !== 'default' && !exports.hasOwnProperty(k)) Object.defineProperty(exports, k, {
		enumerable: true,
		get: function () { return MuiButton$1[k]; }
	});
});
Object.keys(MuiAlertTitle).forEach(function (k) {
	if (k !== 'default' && !exports.hasOwnProperty(k)) Object.defineProperty(exports, k, {
		enumerable: true,
		get: function () { return MuiAlertTitle[k]; }
	});
});
Object.keys(MuiAppBar$1).forEach(function (k) {
	if (k !== 'default' && !exports.hasOwnProperty(k)) Object.defineProperty(exports, k, {
		enumerable: true,
		get: function () { return MuiAppBar$1[k]; }
	});
});
Object.keys(MuiAutocomplete$1).forEach(function (k) {
	if (k !== 'default' && !exports.hasOwnProperty(k)) Object.defineProperty(exports, k, {
		enumerable: true,
		get: function () { return MuiAutocomplete$1[k]; }
	});
});
Object.keys(MuiAvatar).forEach(function (k) {
	if (k !== 'default' && !exports.hasOwnProperty(k)) Object.defineProperty(exports, k, {
		enumerable: true,
		get: function () { return MuiAvatar[k]; }
	});
});
Object.keys(MuiAvatarGroup).forEach(function (k) {
	if (k !== 'default' && !exports.hasOwnProperty(k)) Object.defineProperty(exports, k, {
		enumerable: true,
		get: function () { return MuiAvatarGroup[k]; }
	});
});
Object.keys(MuiBackdrop).forEach(function (k) {
	if (k !== 'default' && !exports.hasOwnProperty(k)) Object.defineProperty(exports, k, {
		enumerable: true,
		get: function () { return MuiBackdrop[k]; }
	});
});
Object.keys(MuiBadge).forEach(function (k) {
	if (k !== 'default' && !exports.hasOwnProperty(k)) Object.defineProperty(exports, k, {
		enumerable: true,
		get: function () { return MuiBadge[k]; }
	});
});
Object.keys(MuiBottomNavigation).forEach(function (k) {
	if (k !== 'default' && !exports.hasOwnProperty(k)) Object.defineProperty(exports, k, {
		enumerable: true,
		get: function () { return MuiBottomNavigation[k]; }
	});
});
Object.keys(MuiBottomNavigationAction).forEach(function (k) {
	if (k !== 'default' && !exports.hasOwnProperty(k)) Object.defineProperty(exports, k, {
		enumerable: true,
		get: function () { return MuiBottomNavigationAction[k]; }
	});
});
Object.keys(MuiBox).forEach(function (k) {
	if (k !== 'default' && !exports.hasOwnProperty(k)) Object.defineProperty(exports, k, {
		enumerable: true,
		get: function () { return MuiBox[k]; }
	});
});
Object.keys(MuiBreadcrumbs).forEach(function (k) {
	if (k !== 'default' && !exports.hasOwnProperty(k)) Object.defineProperty(exports, k, {
		enumerable: true,
		get: function () { return MuiBreadcrumbs[k]; }
	});
});
Object.keys(MuiButtonBase$1).forEach(function (k) {
	if (k !== 'default' && !exports.hasOwnProperty(k)) Object.defineProperty(exports, k, {
		enumerable: true,
		get: function () { return MuiButtonBase$1[k]; }
	});
});
Object.keys(MuiButtonGroup$1).forEach(function (k) {
	if (k !== 'default' && !exports.hasOwnProperty(k)) Object.defineProperty(exports, k, {
		enumerable: true,
		get: function () { return MuiButtonGroup$1[k]; }
	});
});
Object.keys(MuiCard).forEach(function (k) {
	if (k !== 'default' && !exports.hasOwnProperty(k)) Object.defineProperty(exports, k, {
		enumerable: true,
		get: function () { return MuiCard[k]; }
	});
});
Object.keys(MuiCardActionArea).forEach(function (k) {
	if (k !== 'default' && !exports.hasOwnProperty(k)) Object.defineProperty(exports, k, {
		enumerable: true,
		get: function () { return MuiCardActionArea[k]; }
	});
});
Object.keys(MuiCardActions).forEach(function (k) {
	if (k !== 'default' && !exports.hasOwnProperty(k)) Object.defineProperty(exports, k, {
		enumerable: true,
		get: function () { return MuiCardActions[k]; }
	});
});
Object.keys(MuiCardContent).forEach(function (k) {
	if (k !== 'default' && !exports.hasOwnProperty(k)) Object.defineProperty(exports, k, {
		enumerable: true,
		get: function () { return MuiCardContent[k]; }
	});
});
Object.keys(MuiCardHeader).forEach(function (k) {
	if (k !== 'default' && !exports.hasOwnProperty(k)) Object.defineProperty(exports, k, {
		enumerable: true,
		get: function () { return MuiCardHeader[k]; }
	});
});
Object.keys(MuiCardMedia).forEach(function (k) {
	if (k !== 'default' && !exports.hasOwnProperty(k)) Object.defineProperty(exports, k, {
		enumerable: true,
		get: function () { return MuiCardMedia[k]; }
	});
});
Object.keys(MuiCheckbox).forEach(function (k) {
	if (k !== 'default' && !exports.hasOwnProperty(k)) Object.defineProperty(exports, k, {
		enumerable: true,
		get: function () { return MuiCheckbox[k]; }
	});
});
Object.keys(MuiChip$1).forEach(function (k) {
	if (k !== 'default' && !exports.hasOwnProperty(k)) Object.defineProperty(exports, k, {
		enumerable: true,
		get: function () { return MuiChip$1[k]; }
	});
});
Object.keys(MuiCircularProgress$1).forEach(function (k) {
	if (k !== 'default' && !exports.hasOwnProperty(k)) Object.defineProperty(exports, k, {
		enumerable: true,
		get: function () { return MuiCircularProgress$1[k]; }
	});
});
Object.keys(MuiClickAwayListener).forEach(function (k) {
	if (k !== 'default' && !exports.hasOwnProperty(k)) Object.defineProperty(exports, k, {
		enumerable: true,
		get: function () { return MuiClickAwayListener[k]; }
	});
});
Object.keys(MuiCollapse).forEach(function (k) {
	if (k !== 'default' && !exports.hasOwnProperty(k)) Object.defineProperty(exports, k, {
		enumerable: true,
		get: function () { return MuiCollapse[k]; }
	});
});
Object.keys(MuiContainer).forEach(function (k) {
	if (k !== 'default' && !exports.hasOwnProperty(k)) Object.defineProperty(exports, k, {
		enumerable: true,
		get: function () { return MuiContainer[k]; }
	});
});
Object.keys(MuiCssBaseline).forEach(function (k) {
	if (k !== 'default' && !exports.hasOwnProperty(k)) Object.defineProperty(exports, k, {
		enumerable: true,
		get: function () { return MuiCssBaseline[k]; }
	});
});
Object.keys(MuiInputAdornment).forEach(function (k) {
	if (k !== 'default' && !exports.hasOwnProperty(k)) Object.defineProperty(exports, k, {
		enumerable: true,
		get: function () { return MuiInputAdornment[k]; }
	});
});
Object.keys(MuiTextField).forEach(function (k) {
	if (k !== 'default' && !exports.hasOwnProperty(k)) Object.defineProperty(exports, k, {
		enumerable: true,
		get: function () { return MuiTextField[k]; }
	});
});
Object.keys(MuiDialog).forEach(function (k) {
	if (k !== 'default' && !exports.hasOwnProperty(k)) Object.defineProperty(exports, k, {
		enumerable: true,
		get: function () { return MuiDialog[k]; }
	});
});
Object.keys(MuiDialogActions).forEach(function (k) {
	if (k !== 'default' && !exports.hasOwnProperty(k)) Object.defineProperty(exports, k, {
		enumerable: true,
		get: function () { return MuiDialogActions[k]; }
	});
});
Object.keys(MuiDialogContent$1).forEach(function (k) {
	if (k !== 'default' && !exports.hasOwnProperty(k)) Object.defineProperty(exports, k, {
		enumerable: true,
		get: function () { return MuiDialogContent$1[k]; }
	});
});
Object.keys(MuiDialogContentText).forEach(function (k) {
	if (k !== 'default' && !exports.hasOwnProperty(k)) Object.defineProperty(exports, k, {
		enumerable: true,
		get: function () { return MuiDialogContentText[k]; }
	});
});
Object.keys(MuiToolbar).forEach(function (k) {
	if (k !== 'default' && !exports.hasOwnProperty(k)) Object.defineProperty(exports, k, {
		enumerable: true,
		get: function () { return MuiToolbar[k]; }
	});
});
Object.keys(MuiStack).forEach(function (k) {
	if (k !== 'default' && !exports.hasOwnProperty(k)) Object.defineProperty(exports, k, {
		enumerable: true,
		get: function () { return MuiStack[k]; }
	});
});
Object.keys(MuiDialogTitle$1).forEach(function (k) {
	if (k !== 'default' && !exports.hasOwnProperty(k)) Object.defineProperty(exports, k, {
		enumerable: true,
		get: function () { return MuiDialogTitle$1[k]; }
	});
});
Object.keys(MuiDivider).forEach(function (k) {
	if (k !== 'default' && !exports.hasOwnProperty(k)) Object.defineProperty(exports, k, {
		enumerable: true,
		get: function () { return MuiDivider[k]; }
	});
});
Object.keys(MuiDrawer).forEach(function (k) {
	if (k !== 'default' && !exports.hasOwnProperty(k)) Object.defineProperty(exports, k, {
		enumerable: true,
		get: function () { return MuiDrawer[k]; }
	});
});
Object.keys(MuiFab).forEach(function (k) {
	if (k !== 'default' && !exports.hasOwnProperty(k)) Object.defineProperty(exports, k, {
		enumerable: true,
		get: function () { return MuiFab[k]; }
	});
});
Object.keys(MuiFade).forEach(function (k) {
	if (k !== 'default' && !exports.hasOwnProperty(k)) Object.defineProperty(exports, k, {
		enumerable: true,
		get: function () { return MuiFade[k]; }
	});
});
Object.keys(MuiFilledInput).forEach(function (k) {
	if (k !== 'default' && !exports.hasOwnProperty(k)) Object.defineProperty(exports, k, {
		enumerable: true,
		get: function () { return MuiFilledInput[k]; }
	});
});
Object.keys(MuiFormControl).forEach(function (k) {
	if (k !== 'default' && !exports.hasOwnProperty(k)) Object.defineProperty(exports, k, {
		enumerable: true,
		get: function () { return MuiFormControl[k]; }
	});
});
Object.keys(MuiFormControlLabel).forEach(function (k) {
	if (k !== 'default' && !exports.hasOwnProperty(k)) Object.defineProperty(exports, k, {
		enumerable: true,
		get: function () { return MuiFormControlLabel[k]; }
	});
});
Object.keys(MuiFormGroup).forEach(function (k) {
	if (k !== 'default' && !exports.hasOwnProperty(k)) Object.defineProperty(exports, k, {
		enumerable: true,
		get: function () { return MuiFormGroup[k]; }
	});
});
Object.keys(MuiFormHelperText$1).forEach(function (k) {
	if (k !== 'default' && !exports.hasOwnProperty(k)) Object.defineProperty(exports, k, {
		enumerable: true,
		get: function () { return MuiFormHelperText$1[k]; }
	});
});
Object.keys(MuiFormLabel).forEach(function (k) {
	if (k !== 'default' && !exports.hasOwnProperty(k)) Object.defineProperty(exports, k, {
		enumerable: true,
		get: function () { return MuiFormLabel[k]; }
	});
});
Object.keys(MuiGrid).forEach(function (k) {
	if (k !== 'default' && !exports.hasOwnProperty(k)) Object.defineProperty(exports, k, {
		enumerable: true,
		get: function () { return MuiGrid[k]; }
	});
});
Object.keys(MuiGrow).forEach(function (k) {
	if (k !== 'default' && !exports.hasOwnProperty(k)) Object.defineProperty(exports, k, {
		enumerable: true,
		get: function () { return MuiGrow[k]; }
	});
});
Object.keys(MuiIcon).forEach(function (k) {
	if (k !== 'default' && !exports.hasOwnProperty(k)) Object.defineProperty(exports, k, {
		enumerable: true,
		get: function () { return MuiIcon[k]; }
	});
});
Object.keys(MuiImageList).forEach(function (k) {
	if (k !== 'default' && !exports.hasOwnProperty(k)) Object.defineProperty(exports, k, {
		enumerable: true,
		get: function () { return MuiImageList[k]; }
	});
});
Object.keys(MuiImageListItem).forEach(function (k) {
	if (k !== 'default' && !exports.hasOwnProperty(k)) Object.defineProperty(exports, k, {
		enumerable: true,
		get: function () { return MuiImageListItem[k]; }
	});
});
Object.keys(MuiImageListItemBar).forEach(function (k) {
	if (k !== 'default' && !exports.hasOwnProperty(k)) Object.defineProperty(exports, k, {
		enumerable: true,
		get: function () { return MuiImageListItemBar[k]; }
	});
});
Object.keys(MuiInput).forEach(function (k) {
	if (k !== 'default' && !exports.hasOwnProperty(k)) Object.defineProperty(exports, k, {
		enumerable: true,
		get: function () { return MuiInput[k]; }
	});
});
Object.keys(MuiInputBase$1).forEach(function (k) {
	if (k !== 'default' && !exports.hasOwnProperty(k)) Object.defineProperty(exports, k, {
		enumerable: true,
		get: function () { return MuiInputBase$1[k]; }
	});
});
Object.keys(MuiInputLabel).forEach(function (k) {
	if (k !== 'default' && !exports.hasOwnProperty(k)) Object.defineProperty(exports, k, {
		enumerable: true,
		get: function () { return MuiInputLabel[k]; }
	});
});
Object.keys(MuiLinearProgress).forEach(function (k) {
	if (k !== 'default' && !exports.hasOwnProperty(k)) Object.defineProperty(exports, k, {
		enumerable: true,
		get: function () { return MuiLinearProgress[k]; }
	});
});
Object.keys(MuiLink).forEach(function (k) {
	if (k !== 'default' && !exports.hasOwnProperty(k)) Object.defineProperty(exports, k, {
		enumerable: true,
		get: function () { return MuiLink[k]; }
	});
});
Object.keys(MuiList).forEach(function (k) {
	if (k !== 'default' && !exports.hasOwnProperty(k)) Object.defineProperty(exports, k, {
		enumerable: true,
		get: function () { return MuiList[k]; }
	});
});
Object.keys(MuiListItem$1).forEach(function (k) {
	if (k !== 'default' && !exports.hasOwnProperty(k)) Object.defineProperty(exports, k, {
		enumerable: true,
		get: function () { return MuiListItem$1[k]; }
	});
});
Object.keys(MuiListItemAvatar).forEach(function (k) {
	if (k !== 'default' && !exports.hasOwnProperty(k)) Object.defineProperty(exports, k, {
		enumerable: true,
		get: function () { return MuiListItemAvatar[k]; }
	});
});
Object.keys(MuiListItemButton$1).forEach(function (k) {
	if (k !== 'default' && !exports.hasOwnProperty(k)) Object.defineProperty(exports, k, {
		enumerable: true,
		get: function () { return MuiListItemButton$1[k]; }
	});
});
Object.keys(MuiListItemIcon).forEach(function (k) {
	if (k !== 'default' && !exports.hasOwnProperty(k)) Object.defineProperty(exports, k, {
		enumerable: true,
		get: function () { return MuiListItemIcon[k]; }
	});
});
Object.keys(MuiListItemSecondaryAction).forEach(function (k) {
	if (k !== 'default' && !exports.hasOwnProperty(k)) Object.defineProperty(exports, k, {
		enumerable: true,
		get: function () { return MuiListItemSecondaryAction[k]; }
	});
});
Object.keys(MuiListItemText$1).forEach(function (k) {
	if (k !== 'default' && !exports.hasOwnProperty(k)) Object.defineProperty(exports, k, {
		enumerable: true,
		get: function () { return MuiListItemText$1[k]; }
	});
});
Object.keys(MuiListSubheader$1).forEach(function (k) {
	if (k !== 'default' && !exports.hasOwnProperty(k)) Object.defineProperty(exports, k, {
		enumerable: true,
		get: function () { return MuiListSubheader$1[k]; }
	});
});
Object.keys(MuiMenu$1).forEach(function (k) {
	if (k !== 'default' && !exports.hasOwnProperty(k)) Object.defineProperty(exports, k, {
		enumerable: true,
		get: function () { return MuiMenu$1[k]; }
	});
});
Object.keys(MuiMenuItem$1).forEach(function (k) {
	if (k !== 'default' && !exports.hasOwnProperty(k)) Object.defineProperty(exports, k, {
		enumerable: true,
		get: function () { return MuiMenuItem$1[k]; }
	});
});
Object.keys(MuiMenuList).forEach(function (k) {
	if (k !== 'default' && !exports.hasOwnProperty(k)) Object.defineProperty(exports, k, {
		enumerable: true,
		get: function () { return MuiMenuList[k]; }
	});
});
Object.keys(MuiMobileStepper).forEach(function (k) {
	if (k !== 'default' && !exports.hasOwnProperty(k)) Object.defineProperty(exports, k, {
		enumerable: true,
		get: function () { return MuiMobileStepper[k]; }
	});
});
Object.keys(MuiModal).forEach(function (k) {
	if (k !== 'default' && !exports.hasOwnProperty(k)) Object.defineProperty(exports, k, {
		enumerable: true,
		get: function () { return MuiModal[k]; }
	});
});
Object.keys(MuiNativeSelect).forEach(function (k) {
	if (k !== 'default' && !exports.hasOwnProperty(k)) Object.defineProperty(exports, k, {
		enumerable: true,
		get: function () { return MuiNativeSelect[k]; }
	});
});
Object.keys(MuiOutlinedInput).forEach(function (k) {
	if (k !== 'default' && !exports.hasOwnProperty(k)) Object.defineProperty(exports, k, {
		enumerable: true,
		get: function () { return MuiOutlinedInput[k]; }
	});
});
Object.keys(MuiPagination).forEach(function (k) {
	if (k !== 'default' && !exports.hasOwnProperty(k)) Object.defineProperty(exports, k, {
		enumerable: true,
		get: function () { return MuiPagination[k]; }
	});
});
Object.keys(MuiPaginationItem).forEach(function (k) {
	if (k !== 'default' && !exports.hasOwnProperty(k)) Object.defineProperty(exports, k, {
		enumerable: true,
		get: function () { return MuiPaginationItem[k]; }
	});
});
Object.keys(MuiPaper$1).forEach(function (k) {
	if (k !== 'default' && !exports.hasOwnProperty(k)) Object.defineProperty(exports, k, {
		enumerable: true,
		get: function () { return MuiPaper$1[k]; }
	});
});
Object.keys(MuiPopover).forEach(function (k) {
	if (k !== 'default' && !exports.hasOwnProperty(k)) Object.defineProperty(exports, k, {
		enumerable: true,
		get: function () { return MuiPopover[k]; }
	});
});
Object.keys(MuiPopper).forEach(function (k) {
	if (k !== 'default' && !exports.hasOwnProperty(k)) Object.defineProperty(exports, k, {
		enumerable: true,
		get: function () { return MuiPopper[k]; }
	});
});
Object.keys(MuiPortal).forEach(function (k) {
	if (k !== 'default' && !exports.hasOwnProperty(k)) Object.defineProperty(exports, k, {
		enumerable: true,
		get: function () { return MuiPortal[k]; }
	});
});
Object.keys(MuiRadio).forEach(function (k) {
	if (k !== 'default' && !exports.hasOwnProperty(k)) Object.defineProperty(exports, k, {
		enumerable: true,
		get: function () { return MuiRadio[k]; }
	});
});
Object.keys(MuiRadioGroup).forEach(function (k) {
	if (k !== 'default' && !exports.hasOwnProperty(k)) Object.defineProperty(exports, k, {
		enumerable: true,
		get: function () { return MuiRadioGroup[k]; }
	});
});
Object.keys(MuiRating).forEach(function (k) {
	if (k !== 'default' && !exports.hasOwnProperty(k)) Object.defineProperty(exports, k, {
		enumerable: true,
		get: function () { return MuiRating[k]; }
	});
});
Object.keys(MuiScopedCssBaseline).forEach(function (k) {
	if (k !== 'default' && !exports.hasOwnProperty(k)) Object.defineProperty(exports, k, {
		enumerable: true,
		get: function () { return MuiScopedCssBaseline[k]; }
	});
});
Object.keys(MuiSelect$1).forEach(function (k) {
	if (k !== 'default' && !exports.hasOwnProperty(k)) Object.defineProperty(exports, k, {
		enumerable: true,
		get: function () { return MuiSelect$1[k]; }
	});
});
Object.keys(MuiSkeleton).forEach(function (k) {
	if (k !== 'default' && !exports.hasOwnProperty(k)) Object.defineProperty(exports, k, {
		enumerable: true,
		get: function () { return MuiSkeleton[k]; }
	});
});
Object.keys(MuiSlide).forEach(function (k) {
	if (k !== 'default' && !exports.hasOwnProperty(k)) Object.defineProperty(exports, k, {
		enumerable: true,
		get: function () { return MuiSlide[k]; }
	});
});
Object.keys(MuiSlider).forEach(function (k) {
	if (k !== 'default' && !exports.hasOwnProperty(k)) Object.defineProperty(exports, k, {
		enumerable: true,
		get: function () { return MuiSlider[k]; }
	});
});
Object.keys(MuiSnackbar).forEach(function (k) {
	if (k !== 'default' && !exports.hasOwnProperty(k)) Object.defineProperty(exports, k, {
		enumerable: true,
		get: function () { return MuiSnackbar[k]; }
	});
});
Object.keys(MuiSnackbarContent).forEach(function (k) {
	if (k !== 'default' && !exports.hasOwnProperty(k)) Object.defineProperty(exports, k, {
		enumerable: true,
		get: function () { return MuiSnackbarContent[k]; }
	});
});
Object.keys(MuiSpeedDial).forEach(function (k) {
	if (k !== 'default' && !exports.hasOwnProperty(k)) Object.defineProperty(exports, k, {
		enumerable: true,
		get: function () { return MuiSpeedDial[k]; }
	});
});
Object.keys(MuiSpeedDialAction).forEach(function (k) {
	if (k !== 'default' && !exports.hasOwnProperty(k)) Object.defineProperty(exports, k, {
		enumerable: true,
		get: function () { return MuiSpeedDialAction[k]; }
	});
});
Object.keys(MuiSpeedDialIcon).forEach(function (k) {
	if (k !== 'default' && !exports.hasOwnProperty(k)) Object.defineProperty(exports, k, {
		enumerable: true,
		get: function () { return MuiSpeedDialIcon[k]; }
	});
});
Object.keys(MuiStep).forEach(function (k) {
	if (k !== 'default' && !exports.hasOwnProperty(k)) Object.defineProperty(exports, k, {
		enumerable: true,
		get: function () { return MuiStep[k]; }
	});
});
Object.keys(MuiStepButton).forEach(function (k) {
	if (k !== 'default' && !exports.hasOwnProperty(k)) Object.defineProperty(exports, k, {
		enumerable: true,
		get: function () { return MuiStepButton[k]; }
	});
});
Object.keys(MuiStepConnector).forEach(function (k) {
	if (k !== 'default' && !exports.hasOwnProperty(k)) Object.defineProperty(exports, k, {
		enumerable: true,
		get: function () { return MuiStepConnector[k]; }
	});
});
Object.keys(MuiStepContent).forEach(function (k) {
	if (k !== 'default' && !exports.hasOwnProperty(k)) Object.defineProperty(exports, k, {
		enumerable: true,
		get: function () { return MuiStepContent[k]; }
	});
});
Object.keys(MuiStepIcon$1).forEach(function (k) {
	if (k !== 'default' && !exports.hasOwnProperty(k)) Object.defineProperty(exports, k, {
		enumerable: true,
		get: function () { return MuiStepIcon$1[k]; }
	});
});
Object.keys(MuiStepLabel$1).forEach(function (k) {
	if (k !== 'default' && !exports.hasOwnProperty(k)) Object.defineProperty(exports, k, {
		enumerable: true,
		get: function () { return MuiStepLabel$1[k]; }
	});
});
Object.keys(MuiStepper$1).forEach(function (k) {
	if (k !== 'default' && !exports.hasOwnProperty(k)) Object.defineProperty(exports, k, {
		enumerable: true,
		get: function () { return MuiStepper$1[k]; }
	});
});
Object.keys(MuiSwipeableDrawer).forEach(function (k) {
	if (k !== 'default' && !exports.hasOwnProperty(k)) Object.defineProperty(exports, k, {
		enumerable: true,
		get: function () { return MuiSwipeableDrawer[k]; }
	});
});
Object.keys(MuiSwitch).forEach(function (k) {
	if (k !== 'default' && !exports.hasOwnProperty(k)) Object.defineProperty(exports, k, {
		enumerable: true,
		get: function () { return MuiSwitch[k]; }
	});
});
Object.keys(MuiTab).forEach(function (k) {
	if (k !== 'default' && !exports.hasOwnProperty(k)) Object.defineProperty(exports, k, {
		enumerable: true,
		get: function () { return MuiTab[k]; }
	});
});
Object.keys(MuiTabScrollButton).forEach(function (k) {
	if (k !== 'default' && !exports.hasOwnProperty(k)) Object.defineProperty(exports, k, {
		enumerable: true,
		get: function () { return MuiTabScrollButton[k]; }
	});
});
Object.keys(MuiTable).forEach(function (k) {
	if (k !== 'default' && !exports.hasOwnProperty(k)) Object.defineProperty(exports, k, {
		enumerable: true,
		get: function () { return MuiTable[k]; }
	});
});
Object.keys(MuiTableBody).forEach(function (k) {
	if (k !== 'default' && !exports.hasOwnProperty(k)) Object.defineProperty(exports, k, {
		enumerable: true,
		get: function () { return MuiTableBody[k]; }
	});
});
Object.keys(MuiTableCell).forEach(function (k) {
	if (k !== 'default' && !exports.hasOwnProperty(k)) Object.defineProperty(exports, k, {
		enumerable: true,
		get: function () { return MuiTableCell[k]; }
	});
});
Object.keys(MuiTableContainer).forEach(function (k) {
	if (k !== 'default' && !exports.hasOwnProperty(k)) Object.defineProperty(exports, k, {
		enumerable: true,
		get: function () { return MuiTableContainer[k]; }
	});
});
Object.keys(MuiTableFooter).forEach(function (k) {
	if (k !== 'default' && !exports.hasOwnProperty(k)) Object.defineProperty(exports, k, {
		enumerable: true,
		get: function () { return MuiTableFooter[k]; }
	});
});
Object.keys(MuiTableHead).forEach(function (k) {
	if (k !== 'default' && !exports.hasOwnProperty(k)) Object.defineProperty(exports, k, {
		enumerable: true,
		get: function () { return MuiTableHead[k]; }
	});
});
Object.keys(MuiTablePagination).forEach(function (k) {
	if (k !== 'default' && !exports.hasOwnProperty(k)) Object.defineProperty(exports, k, {
		enumerable: true,
		get: function () { return MuiTablePagination[k]; }
	});
});
Object.keys(MuiTableRow).forEach(function (k) {
	if (k !== 'default' && !exports.hasOwnProperty(k)) Object.defineProperty(exports, k, {
		enumerable: true,
		get: function () { return MuiTableRow[k]; }
	});
});
Object.keys(MuiTableSortLabel).forEach(function (k) {
	if (k !== 'default' && !exports.hasOwnProperty(k)) Object.defineProperty(exports, k, {
		enumerable: true,
		get: function () { return MuiTableSortLabel[k]; }
	});
});
Object.keys(MuiTabs).forEach(function (k) {
	if (k !== 'default' && !exports.hasOwnProperty(k)) Object.defineProperty(exports, k, {
		enumerable: true,
		get: function () { return MuiTabs[k]; }
	});
});
Object.keys(MuiTextareaAutosize).forEach(function (k) {
	if (k !== 'default' && !exports.hasOwnProperty(k)) Object.defineProperty(exports, k, {
		enumerable: true,
		get: function () { return MuiTextareaAutosize[k]; }
	});
});
Object.keys(MuiToggleButton$1).forEach(function (k) {
	if (k !== 'default' && !exports.hasOwnProperty(k)) Object.defineProperty(exports, k, {
		enumerable: true,
		get: function () { return MuiToggleButton$1[k]; }
	});
});
Object.keys(MuiToggleButtonGroup).forEach(function (k) {
	if (k !== 'default' && !exports.hasOwnProperty(k)) Object.defineProperty(exports, k, {
		enumerable: true,
		get: function () { return MuiToggleButtonGroup[k]; }
	});
});
Object.keys(MuiTooltip$1).forEach(function (k) {
	if (k !== 'default' && !exports.hasOwnProperty(k)) Object.defineProperty(exports, k, {
		enumerable: true,
		get: function () { return MuiTooltip$1[k]; }
	});
});
Object.keys(MuiTypography).forEach(function (k) {
	if (k !== 'default' && !exports.hasOwnProperty(k)) Object.defineProperty(exports, k, {
		enumerable: true,
		get: function () { return MuiTypography[k]; }
	});
});
Object.keys(MuiZoom).forEach(function (k) {
	if (k !== 'default' && !exports.hasOwnProperty(k)) Object.defineProperty(exports, k, {
		enumerable: true,
		get: function () { return MuiZoom[k]; }
	});
});
Object.keys(hooks).forEach(function (k) {
	if (k !== 'default' && !exports.hasOwnProperty(k)) Object.defineProperty(exports, k, {
		enumerable: true,
		get: function () { return hooks[k]; }
	});
});
