// @elementor/ui/DatePicker/index.esm.js
import React from 'react';
import { DatePicker as DatePicker$1 } from '@mui/x-date-pickers/DatePicker';
import MuiIconButton from '@mui/material/IconButton';
import MuiInputAdornment from '@mui/material/InputAdornment';
import MuiSvgIcon from '@mui/material/SvgIcon';
import MuiTextField from '@mui/material/TextField';

// Modified reason: Material UI Typescript problem with the 'component' prop.
const IconButton = React.forwardRef((props, ref) => {
    return (React.createElement(MuiIconButton, { ...props, ref: ref }));
});

const InputAdornment = React.forwardRef((props, ref) => {
    return (React.createElement(MuiInputAdornment, { ...props, ref: ref }));
});

const SvgIcon = React.forwardRef((props, ref) => {
    return (React.createElement(MuiSvgIcon, { ...props, ref: ref }));
});

const TextField = React.forwardRef((props, ref) => {
    return (React.createElement(MuiTextField, { ...props, ref: ref }));
});

const CalendarIcon = React.forwardRef((props, ref) => {
    return (React.createElement(SvgIcon, { viewBox: "0 0 24 24", ...props, ref: ref },
        React.createElement("path", { fillRule: "evenodd", clipRule: "evenodd", d: "M8 2.25C8.41421 2.25 8.75 2.58579 8.75 3V4.25H15.25V3C15.25 2.58579 15.5858 2.25 16 2.25C16.4142 2.25 16.75 2.58579 16.75 3V4.25H18C19.5188 4.25 20.75 5.48122 20.75 7V19C20.75 20.5188 19.5188 21.75 18 21.75H6C4.48122 21.75 3.25 20.5188 3.25 19V7C3.25 5.48122 4.48122 4.25 6 4.25H7.25V3C7.25 2.58579 7.58579 2.25 8 2.25ZM7.25 5.75H6C5.30964 5.75 4.75 6.30964 4.75 7V10.25H19.25V7C19.25 6.30964 18.6904 5.75 18 5.75H16.75V7C16.75 7.41421 16.4142 7.75 16 7.75C15.5858 7.75 15.25 7.41421 15.25 7V5.75H8.75V7C8.75 7.41421 8.41421 7.75 8 7.75C7.58579 7.75 7.25 7.41421 7.25 7V5.75ZM19.25 11.75H4.75V19C4.75 19.6904 5.30964 20.25 6 20.25H18C18.6904 20.25 19.25 19.6904 19.25 19V11.75Z" })));
});
const DatePicker = React.forwardRef(({ slots = {}, slotProps = {}, ...props }, ref) => {
    return (React.createElement(DatePicker$1, { ...props, ref: ref, slots: {
            clearButton: IconButton,
            inputAdornment: InputAdornment,
            nextIconButton: IconButton,
            openPickerButton: IconButton,
            openPickerIcon: CalendarIcon,
            previousIconButton: IconButton,
            switchViewButton: IconButton,
            textField: TextField,
            ...slots,
        }, slotProps: {
            layout: {
                sx: {
                    '& .MuiDayCalendar-root': {
                        // The calendar layout should always be aligned to the left.
                        direction: 'ltr /* @noflip */',
                    },
                    '& .MuiYearCalendar-root': {
                        // The year selection layout should always be aligned to the left.
                        direction: 'ltr /* @noflip */',
                    },
                },
            },
            ...slotProps,
        } }));
});

export { DatePicker as default };
