// @elementor/ui/TimePicker/index.js
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var React = require('react');
var TimePicker$1 = require('@mui/x-date-pickers/TimePicker');
var MuiIconButton = require('@mui/material/IconButton');
var MuiInputAdornment = require('@mui/material/InputAdornment');
var MuiSvgIcon = require('@mui/material/SvgIcon');
var MuiTextField = require('@mui/material/TextField');

function _interopDefaultLegacy (e) { return e && typeof e === 'object' && 'default' in e ? e : { 'default': e }; }

var React__default = /*#__PURE__*/_interopDefaultLegacy(React);
var MuiIconButton__default = /*#__PURE__*/_interopDefaultLegacy(MuiIconButton);
var MuiInputAdornment__default = /*#__PURE__*/_interopDefaultLegacy(MuiInputAdornment);
var MuiSvgIcon__default = /*#__PURE__*/_interopDefaultLegacy(MuiSvgIcon);
var MuiTextField__default = /*#__PURE__*/_interopDefaultLegacy(MuiTextField);

// Modified reason: Material UI Typescript problem with the 'component' prop.
const IconButton = React__default["default"].forwardRef((props, ref) => {
    return (React__default["default"].createElement(MuiIconButton__default["default"], { ...props, ref: ref }));
});

const InputAdornment = React__default["default"].forwardRef((props, ref) => {
    return (React__default["default"].createElement(MuiInputAdornment__default["default"], { ...props, ref: ref }));
});

const SvgIcon = React__default["default"].forwardRef((props, ref) => {
    return (React__default["default"].createElement(MuiSvgIcon__default["default"], { ...props, ref: ref }));
});

const TextField = React__default["default"].forwardRef((props, ref) => {
    return (React__default["default"].createElement(MuiTextField__default["default"], { ...props, ref: ref }));
});

const ClockIcon = React__default["default"].forwardRef((props, ref) => {
    return (React__default["default"].createElement(SvgIcon, { viewBox: "0 0 24 24", ...props, ref: ref },
        React__default["default"].createElement("path", { fillRule: "evenodd", clipRule: "evenodd", d: "M12 3.75C7.44365 3.75 3.75 7.44365 3.75 12C3.75 16.5563 7.44365 20.25 12 20.25C16.5563 20.25 20.25 16.5563 20.25 12C20.25 7.44365 16.5563 3.75 12 3.75ZM2.25 12C2.25 6.61522 6.61522 2.25 12 2.25C17.3848 2.25 21.75 6.61522 21.75 12C21.75 17.3848 17.3848 21.75 12 21.75C6.61522 21.75 2.25 17.3848 2.25 12ZM12 6.25C12.4142 6.25 12.75 6.58579 12.75 7V11.6893L15.5303 14.4697C15.8232 14.7626 15.8232 15.2374 15.5303 15.5303C15.2374 15.8232 14.7626 15.8232 14.4697 15.5303L11.4697 12.5303C11.329 12.3897 11.25 12.1989 11.25 12V7C11.25 6.58579 11.5858 6.25 12 6.25Z" })));
});
const TimePicker = React__default["default"].forwardRef(({ slots = {}, slotProps = {}, ...props }, ref) => {
    return (React__default["default"].createElement(TimePicker$1.TimePicker, { ...props, ref: ref, slots: {
            clearButton: IconButton,
            inputAdornment: InputAdornment,
            nextIconButton: IconButton,
            openPickerButton: IconButton,
            openPickerIcon: ClockIcon,
            previousIconButton: IconButton,
            textField: TextField,
            ...slots,
        }, slotProps: {
            layout: {
                sx: {
                    // The time controls should always be aligned to the left.
                    direction: 'ltr /* @noflip */',
                },
            },
            actionBar: {
                sx: {
                    // The action bar should be aligned according to the general direction (flipped by Material-UI in RTL mode).
                    direction: 'ltr',
                },
            },
            ...slotProps,
        } }));
});

exports["default"] = TimePicker;
