// @elementor/ui/Link/index.js
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var React = require('react');
var MuiLink = require('@mui/material/Link');

function _interopDefaultLegacy (e) { return e && typeof e === 'object' && 'default' in e ? e : { 'default': e }; }

var React__default = /*#__PURE__*/_interopDefaultLegacy(React);
var MuiLink__default = /*#__PURE__*/_interopDefaultLegacy(MuiLink);

const LINK_PSEUDO_SELECTORS = '&:hover,&:focus,&:active,&:visited';

/**
 * All supported colors for the Link component that can be passed as a string (without using primary.main, secondary.main, etc.).
 *
 * @see https://github.com/mui/material-ui/blob/f3945c1227afc13fbf732e66875095a793af2f10/packages/mui-material/src/Typography/Typography.js#L82
 */
const colorTransformations = {
    primary: 'primary.main',
    textPrimary: 'text.primary',
    secondary: 'secondary.main',
    textSecondary: 'text.secondary',
    error: 'error.main',
};
// The sx prop does not support the color values of the Link component, so we need to transform them to the correct format.
const convertToSxSupportedColor = (color) => {
    // Temporary customization until the pink color will be replaced with purple.
    if (color === 'primary.main' || color === 'primary') {
        return '#C00BB9';
    }
    if (color === 'global.main') {
        return '#17929B';
    }
    return colorTransformations[color] || color;
};
// Modified reason: the Link component does not provide default style for :hover, therefore global CSS can override the link color.
const Link = React__default["default"].forwardRef((inProps, ref) => {
    const { sx = {}, ...props } = inProps;
    const sxSupportedColor = convertToSxSupportedColor(props.color);
    return (React__default["default"].createElement(MuiLink__default["default"], { ...props, color: sxSupportedColor, sx: {
            // Setting the pseudo-classes color by default to prevent global CSS from overriding the link color on :hover.
            [LINK_PSEUDO_SELECTORS]: {
                color: sxSupportedColor,
            },
            ...sx,
        }, ref: ref }));
});
Link.defaultProps = {
    color: 'primary.main',
};

exports["default"] = Link;
Object.keys(MuiLink).forEach(function (k) {
	if (k !== 'default' && !exports.hasOwnProperty(k)) Object.defineProperty(exports, k, {
		enumerable: true,
		get: function () { return MuiLink[k]; }
	});
});
